ccl/testccl/workload/schemachange: TestWorkload failed

[(ccl/testccl/workload/schemachange).TestWorkload failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2846558&tab=buildLog) on [master@6e5dfc46eab59f7fd6e77ee5dbab5bac715330de](https://github.com/cockroachdb/cockroach/commits/6e5dfc46eab59f7fd6e77ee5dbab5bac715330de):

```
1, 29, zones, 5
50, 0, public, 29
51, 0, public, 29
52, 0, public, 29
52, 0, schema16, 60
52, 29, _enum1, 54
52, 29, enum1, 53
52, 29, seq3, 55
52, 60, table19, 61
I210404 05:30:18.487814 7115 9@util/log/event_log.go:32  [n1,client=127.0.0.1:45550,hostssl,user=root,intExec=backup-lookup-tenant] 3133 ={"Timestamp":1617514218484638714,"EventType":"query_execute","Statement":"SELECT id, active, info FROM \"\".system.tenants","User":"root","ApplicationName":"$ internal-backup-lookup-tenant","ExecMode":"exec-internal","Age":0.923304,"FullTableScan":true}
I210404 05:30:18.489136 7116 9@util/log/event_log.go:32  [n1,client=127.0.0.1:45550,hostssl,user=root,intExec=job-row-insert] 3134 ={"Timestamp":1617514218484638714,"EventType":"query_execute","Statement":"INSERT INTO \"\".system.jobs(id, status, payload, progress, claim_session_id, claim_instance_id) VALUES ($1, $2, $3, $4, $5, $6)","User":"root","ApplicationName":"$ internal-job-row-insert","PlaceholderValues":["646983904421675009","'running'","'\\x0a364241434b555020444154414241534520736368656d616368616e676520544f20276e6f64656c6f63616c3a2f2f302f6261636b7570271204726f6f743206343536373c3d52360a00120a08d0afb9dfc993a4b9161a146e6f64656c6f63616c3a2f2f302f6261636b75703a109c541322b7524bc5b6227f127ab9f9ef'","'\\x109fe7a583eee3ef025200'","'\\x044f16166f484655b596d0e84fbcafa7'","1"],"ExecMode":"exec-internal","NumRows":1,"Age":1.026931}
I210404 05:30:18.503424 7118 9@util/log/event_log.go:32  [n1,client=127.0.0.1:45550,hostssl,user=root,intExec=protectedts-protect] 3135 ={"Timestamp":1617514218484638714,"EventType":"query_execute","Statement":"WITH current_meta AS (SELECT version, num_records, num_spans, total_bytes FROM \"\".system.protected_ts_meta UNION ALL SELECT 0 AS version, 0 AS num_records, 0 AS num_spans, 0 AS total_bytes ORDER BY version DESC LIMIT 1), checks AS (SELECT new_version, new_num_records, new_num_spans, new_total_bytes, (((($1 > 0) AND (new_num_spans > $1)) OR (($2 > 0) AND (new_total_bytes > $2))) OR EXISTS (SELECT * FROM \"\".system.protected_ts_records WHERE id = $4)) AS failed FROM (SELECT version + 1 AS new_version, num_records + 1 AS new_num_records, num_spans + $3 AS new_num_spans, ((total_bytes + length($9)) + length($6)) + COALESCE(length($7), 0) AS new_total_bytes FROM \"\".\"\".current_meta)), updated_meta AS (UPSERT INTO \"\".system.protected_ts_meta(version, num_records, num_spans, total_bytes) (SELECT new_version, new_num_records, new_num_spans, new_total_bytes FROM \"\".\"\".checks WHERE NOT failed) RETURNING version, num_records, num_spans,
I210404 05:30:18.503424 7118 9@util/log/event_log.go:32  [n1,client=127.0.0.1:45550,hostssl,user=root,intExec=protectedts-protect] 3135 | total_bytes), new_record AS (INSERT INTO \"\".system.protected_ts_records(id, ts, meta_type, meta, num_spans, spans) (SELECT $4, $5, $6, $7, $8, $9 WHERE NOT EXISTS (SELECT * FROM \"\".\"\".checks WHERE failed)) RETURNING id) SELECT failed, num_spans AS prev_spans, total_bytes AS prev_total_bytes, version AS prev_version FROM \"\".\"\".checks, \"\".\"\".current_meta","User":"node","ApplicationName":"$ internal-protectedts-protect","PlaceholderValues":["32768","1048576","2","'\\x9c541322b7524bc5b6227f127ab9f9ef'","'1617514218484815824.0000000000'","'jobs'","'\\x363436393833393034343231363735303039'","2","'\\x3a081a02bf892202bf8a3a081a02c5892202c592'"],"ExecMode":"exec-internal","NumRows":1,"SQLSTATE":"XXUUU","ErrorText":"synchronous ieResultReader closed","Age":5.45703,"FullTableScan":true}
I210404 05:30:18.505887 7080 9@util/log/event_log.go:32  [n1,client=127.0.0.1:45550,hostssl,user=root,intExec=log-job] 3136 ={"Timestamp":1617514218504486633,"EventType":"query_execute","Statement":"SELECT status, payload, progress, claim_session_id FROM \"\".system.jobs WHERE id = $1 FOR UPDATE","User":"root","ApplicationName":"$ internal-log-job","PlaceholderValues":["646983904421675009"],"ExecMode":"exec-internal","NumRows":1,"Age":1.371144}
I210404 05:30:18.507403 7082 9@util/log/event_log.go:32  [n1,client=127.0.0.1:45550,hostssl,user=root,intExec=job-update] 3137 ={"Timestamp":1617514218504486633,"EventType":"query_execute","Statement":"UPDATE \"\".system.jobs SET status = $2, payload = $3 WHERE id = $1","User":"root","ApplicationName":"$ internal-job-update","PlaceholderValues":["646983904421675009","'running'","'\\x0a364241434b555020444154414241534520736368656d616368616e676520544f20276e6f64656c6f63616c3a2f2f302f6261636b7570271204726f6f7418898ea783eee3ef023206343536373c3d52360a00120a08d0afb9dfc993a4b9161a146e6f64656c6f63616c3a2f2f302f6261636b75703a109c541322b7524bc5b6227f127ab9f9ef'"],"ExecMode":"exec-internal","NumRows":1,"Age":1.372197}
I210404 05:30:18.508130 7203 jobs/registry.go:1175  [n1] 3138  BACKUP job 646983904421675009: stepping through state running with error: <nil>
I210404 05:30:18.511329 7204 9@util/log/event_log.go:32  [n1,job=646983904421675009,intExec=protectedts-GetRecord] 3139 ={"Timestamp":1617514218508273024,"EventType":"query_execute","Statement":"SELECT id, ts, meta_type, meta, spans, verified FROM \"\".system.protected_ts_records WHERE id = $1","User":"node","ApplicationName":"$ internal-protectedts-GetRecord","PlaceholderValues":["'\\x9c541322b7524bc5b6227f127ab9f9ef'"],"ExecMode":"exec-internal","NumRows":1,"Age":3.025303}
I210404 05:30:18.512945 7152 9@util/log/event_log.go:32  [intExec=protectedts-GetMetadata] 3140 ={"Timestamp":1617514218511565953,"EventType":"query_execute","Statement":"WITH current_meta AS (SELECT version, num_records, num_spans, total_bytes FROM \"\".system.protected_ts_meta UNION ALL SELECT 0 AS version, 0 AS num_records, 0 AS num_spans, 0 AS total_bytes ORDER BY version DESC LIMIT 1) SELECT version, num_records, num_spans, total_bytes FROM \"\".\"\".current_meta","User":"node","ApplicationName":"$ internal-protectedts-GetMetadata","ExecMode":"exec-internal","NumRows":1,"Age":1.338908,"FullTableScan":true}
I210404 05:30:18.514029 7153 9@util/log/event_log.go:32  [intExec=protectedts-GetMetadata] 3141 ={"Timestamp":1617514218511565953,"EventType":"query_execute","Statement":"WITH current_meta AS (SELECT version, num_records, num_spans, total_bytes FROM \"\".system.protected_ts_meta UNION ALL SELECT 0 AS version, 0 AS num_records, 0 AS num_spans, 0 AS total_bytes ORDER BY version DESC LIMIT 1) SELECT version, num_records, num_spans, total_bytes FROM \"\".\"\".current_meta","User":"node","ApplicationName":"$ internal-protectedts-GetMetadata","ExecMode":"exec-internal","NumRows":1,"Age":0.975372,"FullTableScan":true}
I210404 05:30:18.514935 7218 9@util/log/event_log.go:32  [intExec=protectedts-GetRecords] 3142 ={"Timestamp":1617514218511565953,"EventType":"query_execute","Statement":"SELECT id, ts, meta_type, meta, spans, verified FROM \"\".system.protected_ts_records","User":"node","ApplicationName":"$ internal-protectedts-GetRecords","ExecMode":"exec-internal","NumRows":1,"Age":0.809535,"FullTableScan":true}
I210404 05:30:18.516470 7219 9@util/log/event_log.go:32  [n1,job=646983904421675009,intExec=protectedts-MarkVerified] 3143 ={"Timestamp":1617514218515061803,"EventType":"query_execute","Statement":"UPDATE \"\".system.protected_ts_records SET verified = true WHERE id = $1 RETURNING true","User":"node","ApplicationName":"$ internal-protectedts-MarkVerified","PlaceholderValues":["'\\x9c541322b7524bc5b6227f127ab9f9ef'"],"ExecMode":"exec-internal","NumRows":1,"Age":1.386197}
I210404 05:30:18.525956 7231 ccl/backupccl/backup_processor.go:315  [n1,job=646983904421675009] 3144  sending ExportRequest for span /Table/55/{1-2} (attempt 1, priority normal)
I210404 05:30:18.526196 7231 ccl/storageccl/export.go:98  [n1,s1,r38/1:/Table/{55-61}] 3145  export [/Table/55/1,/Table/55/2)
I210404 05:30:18.526574 7227 ccl/backupccl/backup_processor.go:315  [n1,job=646983904421675009] 3146  sending ExportRequest for span /Table/61/1{-0} (attempt 1, priority normal)
I210404 05:30:18.526777 7227 ccl/storageccl/export.go:98  [n1,s1,r39/1:/{Table/61-Max}] 3147  export [/Table/61/1,/Table/61/10)
I210404 05:30:18.529762 7207 9@util/log/event_log.go:32  [n1,job=646983904421675009,intExec=log-job] 3148 ={"Timestamp":1617514218527993912,"EventType":"query_execute","Statement":"SELECT status, payload, progress, claim_session_id FROM \"\".system.jobs WHERE id = $1 FOR UPDATE","User":"root","ApplicationName":"$ internal-log-job","PlaceholderValues":["646983904421675009"],"ExecMode":"exec-internal","NumRows":1,"Age":1.722509}
I210404 05:30:18.531596 7209 9@util/log/event_log.go:32  [n1,job=646983904421675009,intExec=job-update] 3149 ={"Timestamp":1617514218527993912,"EventType":"query_execute","Statement":"UPDATE \"\".system.jobs SET progress = $2 WHERE id = $1","User":"root","ApplicationName":"$ internal-job-update","PlaceholderValues":["646983904421675009","'\\x0d0000003f10fab9a883eee3ef025200'"],"ExecMode":"exec-internal","NumRows":1,"Age":1.683151}
I210404 05:30:18.538144 7177 9@util/log/event_log.go:32  [n1,job=646983904421675009,intExec=log-job] 3150 ={"Timestamp":1617514218536451875,"EventType":"query_execute","Statement":"SELECT status, payload, progress, claim_session_id FROM \"\".system.jobs WHERE id = $1 FOR UPDATE","User":"root","ApplicationName":"$ internal-log-job","PlaceholderValues":["646983904421675009"],"ExecMode":"exec-internal","NumRows":1,"Age":1.659558}
I210404 05:30:18.540160 7179 9@util/log/event_log.go:32  [n1,job=646983904421675009,intExec=job-update] 3151 ={"Timestamp":1617514218536451875,"EventType":"query_execute","Statement":"UPDATE \"\".system.jobs SET progress = $2 WHERE id = $1","User":"root","ApplicationName":"$ internal-job-update","PlaceholderValues":["646983904421675009","'\\x0d0000803f1084fca883eee3ef025200'"],"ExecMode":"exec-internal","NumRows":1,"Age":1.547719}
I210404 05:30:18.562900 7234 9@util/log/event_log.go:32  [n1,job=646983904421675009,intExec=get-table-statistics] 3152 ={"Timestamp":1617514218562205077,"EventType":"query_execute","Statement":"SELECT \"tableID\", \"statisticID\", name, \"columnIDs\", \"createdAt\", \"rowCount\", \"distinctCount\", \"nullCount\", histogram FROM \"\".system.table_statistics WHERE \"tableID\" = $1 ORDER BY \"createdAt\" DESC","User":"root","ApplicationName":"$ internal-get-table-statistics","PlaceholderValues":["55"],"ExecMode":"exec-internal","Age":16.097494,"TxnCounter":1}
I210404 05:30:18.572179 7165 9@util/log/event_log.go:32  [n1,job=646983904421675009,intExec=protectedts-Release] 3153 ={"Timestamp":1617514218566433367,"EventType":"query_execute","Statement":"WITH current_meta AS (SELECT version, num_records, num_spans, total_bytes FROM \"\".system.protected_ts_meta UNION ALL SELECT 0 AS version, 0 AS num_records, 0 AS num_spans, 0 AS total_bytes ORDER BY version DESC LIMIT 1), record AS (SELECT id, num_spans AS record_spans, (length(spans) + length(meta_type)) + COALESCE(length(meta), 0) AS record_bytes FROM \"\".system.protected_ts_records WHERE id = $1), updated_meta AS (UPSERT INTO \"\".system.protected_ts_meta(version, num_records, num_spans, total_bytes) (SELECT version, num_records, num_spans, total_bytes FROM (SELECT version + 1 AS version, num_records - 1 AS num_records, num_spans - record_spans AS num_spans, total_bytes - record_bytes AS total_bytes FROM \"\".\"\".current_meta RIGHT JOIN \"\".\"\".record ON true)) RETURNING 1) DELETE FROM \"\".system.protected_ts_records AS r WHERE EXISTS (SELECT NULL FROM \"\".\"\".record WHERE r.id = record.id) RETURNING NULL","User":"node","Appli
I210404 05:30:18.572179 7165 9@util/log/event_log.go:32  [n1,job=646983904421675009,intExec=protectedts-Release] 3153 |cationName":"$ internal-protectedts-Release","PlaceholderValues":["'\\x9c541322b7524bc5b6227f127ab9f9ef'"],"ExecMode":"exec-internal","NumRows":1,"Age":5.679461,"FullTableScan":true}
I210404 05:30:18.574900 7193 9@util/log/event_log.go:32  [n1,job=646983904421675009,intExec=lookup-schedule-info] 3154 ={"Timestamp":1617514218573342851,"EventType":"query_execute","Statement":"SELECT created_by_id FROM \"\".system.jobs WHERE (id = $1) AND (created_by_type = $2)","User":"node","ApplicationName":"$ internal-lookup-schedule-info","PlaceholderValues":["646983904421675009","'crdb_schedule'"],"ExecMode":"exec-internal","Age":1.527936}
I210404 05:30:18.575019 7203 jobs/registry.go:1175  [n1] 3155  BACKUP job 646983904421675009: stepping through state succeeded with error: <nil>
I210404 05:30:18.576422 7194 9@util/log/event_log.go:32  [n1,intExec=log-job] 3156 ={"Timestamp":1617514218575068011,"EventType":"query_execute","Statement":"SELECT status, payload, progress, claim_session_id FROM \"\".system.jobs WHERE id = $1 FOR UPDATE","User":"root","ApplicationName":"$ internal-log-job","PlaceholderValues":["646983904421675009"],"ExecMode":"exec-internal","NumRows":1,"Age":1.329504}
I210404 05:30:18.578138 7196 9@util/log/event_log.go:32  [n1,intExec=job-update] 3157 ={"Timestamp":1617514218575068011,"EventType":"query_execute","Statement":"UPDATE \"\".system.jobs SET status = $2, payload = $3, progress = $4 WHERE id = $1","User":"root","ApplicationName":"$ internal-job-update","PlaceholderValues":["646983904421675009","'succeeded'","'\\x0a364241434b555020444154414241534520736368656d616368616e676520544f20276e6f64656c6f63616c3a2f2f302f6261636b7570271204726f6f7418898ea783eee3ef02208fb5ab83eee3ef023206343536373c3d52360a00120a08d0afb9dfc993a4b9161a146e6f64656c6f63616c3a2f2f302f6261636b75703a109c541322b7524bc5b6227f127ab9f9ef'","'\\x0d0000803f10dca9ab83eee3ef025200'"],"ExecMode":"exec-internal","NumRows":1,"Age":1.569038}
I210404 05:30:18.579925 4618 9@util/log/event_log.go:32  [n1,client=127.0.0.1:45550,hostssl,user=root] 3158 ={"Timestamp":1617514218484638714,"EventType":"query_execute","Statement":"BACKUP DATABASE schemachange TO 'nodelocal://0/backup'","User":"root","ExecMode":"exec","NumRows":1,"Age":95.329544,"TxnCounter":7}
    schema_change_external_test.go:81: backup in /go/src/github.com/cockroachdb/cockroach/artifacts/TestWorkload667492225
I210404 05:30:18.580127 10 testutils/testcluster/testcluster.go:101  [-] 3159  TestCluster quiescing nodes
W210404 05:30:18.580232 1907 sql/sqlliveness/slinstance/slinstance.go:183  [n3] 3160  exiting heartbeat loop
W210404 05:30:18.580481 1908 jobs/registry.go:729  [-] 3161  canceling all adopted jobs due to stopper quiescing
W210404 05:30:18.580495 2052 kv/kvserver/closedts/sidetransport/receiver.go:125  [n3] 3162  closed timestamps side-transport connection dropped from node: 1
W210404 05:30:18.580739 396 jobs/registry.go:729  [-] 3163  canceling all adopted jobs due to stopper quiescing
E210404 05:30:18.581004 7184 kv/kvclient/kvcoord/txn_interceptor_committer.go:452  [-] 3164  making txn commit explicit failed for "unnamed" meta={id=a0ce0c7c pri=0.00191801 epo=0 ts=1617514218.575068011,0 min=1617514218.575068011,0 seq=6} lock=true stat=STAGING rts=1617514218.575068011,0 wto=false gul=1617514219.075068011,0 int=1 ifw=5: result is ambiguous (server shutdown)
W210404 05:30:18.581162 7167 kv/kvserver/intentresolver/intent_resolver.go:557  [-] 3165  failed to cleanup transaction intents: failed to resolve intents: node unavailable; try another peer
W210404 05:30:18.581181 395 sql/sqlliveness/slinstance/slinstance.go:183  [n1] 3166  exiting heartbeat loop
W210404 05:30:18.581450 1514 jobs/registry.go:729  [-] 3167  canceling all adopted jobs due to stopper quiescing
W210404 05:30:18.581832 1513 sql/sqlliveness/slinstance/slinstance.go:183  [n2] 3169  exiting heartbeat loop
W210404 05:30:18.581580 2573 kv/kvserver/closedts/sidetransport/receiver.go:125  [n2] 3168  closed timestamps side-transport connection dropped from node: 1
--- FAIL: TestWorkload (3.02s)

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=TestWorkload PKG=./pkg/ccl/testccl/workload/schemachange TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestWorkload.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>


Jira issue: CRDB-6414

[(ccl/testccl/workload/schemachange).TestWorkload failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2854447&tab=buildLog) on [master@e7f724493872d75ff1124959d2fba1aa3cdf34e1](https://github.com/cockroachdb/cockroach/commits/e7f724493872d75ff1124959d2fba1aa3cdf34e1):

```
1, 29, tenants, 8
1, 29, ui, 14
1, 29, users, 4
1, 29, web_sessions, 19
1, 29, zones, 5
50, 0, public, 29
51, 0, public, 29
52, 0, public, 29
52, 0, schema211, 104
52, 0, schema246, 113
52, 0, schema339, 134
52, 29, _crdb_internal_region, 56
52, 29, _enum121, 73
52, 29, _enum296, 126
52, 29, _enum62, 66
52, 29, crdb_internal_region, 55
52, 29, enum121, 72
52, 29, enum296, 125
52, 29, enum62, 65
52, 113, _enum277, 120
52, 113, _enum306, 130
52, 113, enum277, 119
52, 113, enum306, 129
I210406 23:50:26.950801 36235 9@util/log/event_log.go:32  [n1,client=127.0.0.1:39098,hostssl,user=root,intExec=backup-lookup-tenant] 7973 ={"Timestamp":1617753026935715455,"EventType":"query_execute","Statement":"SELECT id, active, info FROM \"\".system.tenants","User":"root","ApplicationName":"$ internal-backup-lookup-tenant","ExecMode":"exec-internal","Age":2.822038,"FullTableScan":true}
I210406 23:50:26.958552 36236 9@util/log/event_log.go:32  [n1,client=127.0.0.1:39098,hostssl,user=root,intExec=job-row-insert] 7974 ={"Timestamp":1617753026935715455,"EventType":"query_execute","Statement":"INSERT INTO \"\".system.jobs(id, status, payload, progress, claim_session_id, claim_instance_id) VALUES ($1, $2, $3, $4, $5, $6)","User":"root","ApplicationName":"$ internal-job-row-insert","PlaceholderValues":["647766431993757697","'running'","'\\x0a364241434b555020444154414241534520736368656d616368616e676520544f20276e6f64656c6f63616c3a2f2f302f6261636b7570271204726f6f74321334373841424849687177787d7e81018201860152240a00120a08d988ddc6e8b9dab9161a146e6f64656c6f63616c3a2f2f302f6261636b7570'","'\\x10a3f784d4e7eaef025200'","'\\x9f52247ada0c432990ed66219ad78aa0'","1"],"ExecMode":"exec-internal","NumRows":1,"Age":7.208893}
I210406 23:50:26.981486 36081 9@util/log/event_log.go:32  [n1,client=127.0.0.1:39098,hostssl,user=root,intExec=log-job] 7975 ={"Timestamp":1617753026976389001,"EventType":"query_execute","Statement":"SELECT status, payload, progress, claim_session_id FROM \"\".system.jobs WHERE id = $1 FOR UPDATE","User":"root","ApplicationName":"$ internal-log-job","PlaceholderValues":["647766431993757697"],"ExecMode":"exec-internal","NumRows":1,"Age":5.026134}
I210406 23:50:26.989256 36307 9@util/log/event_log.go:32  [n1,client=127.0.0.1:39098,hostssl,user=root,intExec=job-update] 7976 ={"Timestamp":1617753026976389001,"EventType":"query_execute","Statement":"UPDATE \"\".system.jobs SET status = $2, payload = $3 WHERE id = $1","User":"root","ApplicationName":"$ internal-job-update","PlaceholderValues":["647766431993757697","'running'","'\\x0a364241434b555020444154414241534520736368656d616368616e676520544f20276e6f64656c6f63616c3a2f2f302f6261636b7570271204726f6f7418d7de87d4e7eaef02321334373841424849687177787d7e81018201860152240a00120a08d988ddc6e8b9dab9161a146e6f64656c6f63616c3a2f2f302f6261636b7570'"],"ExecMode":"exec-internal","NumRows":1,"Age":7.447977}
I210406 23:50:26.991481 36310 jobs/registry.go:1175  [n1] 7977  BACKUP job 647766431993757697: stepping through state running with error: <nil>
I210406 23:50:27.069296 36359 9@util/log/event_log.go:32  [n1,job=647766431993757697,intExec=lookup-schedule-info] 7978 ={"Timestamp":1617753027065527261,"EventType":"query_execute","Statement":"SELECT created_by_id FROM \"\".system.jobs WHERE (id = $1) AND (created_by_type = $2)","User":"node","ApplicationName":"$ internal-lookup-schedule-info","PlaceholderValues":["647766431993757697","'crdb_schedule'"],"ExecMode":"exec-internal","Age":3.696432}
I210406 23:50:27.069528 36310 jobs/registry.go:1175  [n1] 7979  BACKUP job 647766431993757697: stepping through state succeeded with error: <nil>
I210406 23:50:27.072453 36360 9@util/log/event_log.go:32  [n1,intExec=log-job] 7980 ={"Timestamp":1617753027069585870,"EventType":"query_execute","Statement":"SELECT status, payload, progress, claim_session_id FROM \"\".system.jobs WHERE id = $1 FOR UPDATE","User":"root","ApplicationName":"$ internal-log-job","PlaceholderValues":["647766431993757697"],"ExecMode":"exec-internal","NumRows":1,"Age":2.771725}
I210406 23:50:27.076362 36362 9@util/log/event_log.go:32  [n1,intExec=job-update] 7981 ={"Timestamp":1617753027069585870,"EventType":"query_execute","Statement":"UPDATE \"\".system.jobs SET status = $2, payload = $3, progress = $4 WHERE id = $1","User":"root","ApplicationName":"$ internal-job-update","PlaceholderValues":["647766431993757697","'succeeded'","'\\x0a364241434b555020444154414241534520736368656d616368616e676520544f20276e6f64656c6f63616c3a2f2f302f6261636b7570271204726f6f7418d7de87d4e7eaef0220a3a68dd4e7eaef02321334373841424849687177787d7e81018201860152240a00120a08d988ddc6e8b9dab9161a146e6f64656c6f63616c3a2f2f302f6261636b7570'","'\\x0d0000803f10928d8dd4e7eaef025200'"],"ExecMode":"exec-internal","NumRows":1,"Age":3.468852}
I210406 23:50:27.079021 5552 9@util/log/event_log.go:32  [n1,client=127.0.0.1:39098,hostssl,user=root] 7982 ={"Timestamp":1617753026935715455,"EventType":"query_execute","Statement":"BACKUP DATABASE schemachange TO 'nodelocal://0/backup'","User":"root","ExecMode":"exec","NumRows":1,"Age":143.39032,"TxnCounter":7}
    schema_change_external_test.go:81: backup in /go/src/github.com/cockroachdb/cockroach/artifacts/TestWorkload910279780
I210406 23:50:27.079560 66 testutils/testcluster/testcluster.go:101  [-] 7983  TestCluster quiescing nodes
W210406 23:50:27.079953 567 sql/sqlliveness/slinstance/slinstance.go:183  [n1] 7984  exiting heartbeat loop
E210406 23:50:27.080068 36131 kv/kvclient/kvcoord/txn_interceptor_committer.go:452  [-] 7985  making txn commit explicit failed for "unnamed" meta={id=e908301a pri=0.00032997 epo=0 ts=1617753027.069585870,0 min=1617753027.069585870,0 seq=6} lock=true stat=STAGING rts=1617753027.069585870,0 wto=false gul=1617753027.569585870,0 int=1 ifw=5: result is ambiguous (server shutdown)
W210406 23:50:27.080626 568 jobs/registry.go:729  [-] 7986  canceling all adopted jobs due to stopper quiescing
W210406 23:50:27.080878 11877 kv/kvserver/closedts/sidetransport/receiver.go:125  [n1] 7987  closed timestamps side-transport connection dropped from node: 3
W210406 23:50:27.080882 2131 jobs/registry.go:729  [-] 7988  canceling all adopted jobs due to stopper quiescing
W210406 23:50:27.081046 7879 kv/kvserver/closedts/sidetransport/receiver.go:125  [n1] 7989  closed timestamps side-transport connection dropped from node: 2
W210406 23:50:27.081483 1631 sql/sqlliveness/slinstance/slinstance.go:183  [n2] 7990  exiting heartbeat loop
W210406 23:50:27.081831 2130 sql/sqlliveness/slinstance/slinstance.go:183  [n3] 7991  exiting heartbeat loop
W210406 23:50:27.082734 7870 kv/kvserver/closedts/sidetransport/receiver.go:125  [n3] 7992  closed timestamps side-transport connection dropped from node: 2
W210406 23:50:27.082930 2468 kv/kvserver/closedts/sidetransport/receiver.go:125  [n3] 7993  closed timestamps side-transport connection dropped from node: 1
W210406 23:50:27.083990 1632 jobs/registry.go:729  [-] 7994  canceling all adopted jobs due to stopper quiescing
W210406 23:50:27.084280 11821 kv/kvserver/closedts/sidetransport/receiver.go:125  [n2] 7995  closed timestamps side-transport connection dropped from node: 3
W210406 23:50:27.085010 1397 2@gossip/gossip.go:1491  [n2] 7996  no incoming or outgoing connections
W210406 23:50:27.085154 2292 kv/kvserver/closedts/sidetransport/receiver.go:125  [n2] 7997  closed timestamps side-transport connection dropped from node: 1
--- FAIL: TestWorkload (45.23s)

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestWorkload PKG=./pkg/ccl/testccl/workload/schemachange TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestWorkload.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(ccl/testccl/workload/schemachange).TestWorkload failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2855122&tab=buildLog) on [master@651184b3cfbd7c6d336bb49e7152e3d48ea06416](https://github.com/cockroachdb/cockroach/commits/651184b3cfbd7c6d336bb49e7152e3d48ea06416):

```
52, 79, _enum159, 101
52, 79, enum159, 100
52, 79, seq249, 124
52, 83, _enum148, 95
52, 83, enum148, 94
52, 93, _enum197, 111
52, 93, enum197, 110
52, 119, _enum236, 121
52, 119, enum236, 120
I210407 05:37:01.196350 24053 9@util/log/event_log.go:32  [n1,client=127.0.0.1:43592,hostssl,user=root,intExec=backup-lookup-tenant] 7872 ={"Timestamp":1617773821188946861,"EventType":"query_execute","Statement":"SELECT id, active, info FROM \"\".system.tenants","User":"root","ApplicationName":"$ internal-backup-lookup-tenant","ExecMode":"exec-internal","Age":1.104018,"FullTableScan":true}
I210407 05:37:01.198019 24054 9@util/log/event_log.go:32  [n1,client=127.0.0.1:43592,hostssl,user=root,intExec=job-row-insert] 7873 ={"Timestamp":1617773821188946861,"EventType":"query_execute","Statement":"INSERT INTO \"\".system.jobs(id, status, payload, progress, claim_session_id, claim_instance_id) VALUES ($1, $2, $3, $4, $5, $6)","User":"root","ApplicationName":"$ internal-job-row-insert","PlaceholderValues":["647834570576920577","'running'","'\\x0a364241434b555020444154414241534520736368656d616368616e676520544f20276e6f64656c6f63616c3a2f2f302f6261636b7570271204726f6f74321634434446474f535d5e5f60616465666e6f717778797c52360a00120a08c2c29e978197dfb9161a146e6f64656c6f63616c3a2f2f302f6261636b75703a1035450b3dac6a42f8963845f6aeda13ed'","'\\x10d3b6c18fb5ebef025200'","'\\xb09f68a9d1e4453cba30ed22919c53d0'","1"],"ExecMode":"exec-internal","NumRows":1,"Age":1.39357}
I210407 05:37:01.213629 24056 9@util/log/event_log.go:32  [n1,client=127.0.0.1:43592,hostssl,user=root,intExec=protectedts-protect] 7874 ={"Timestamp":1617773821188946861,"EventType":"query_execute","Statement":"WITH current_meta AS (SELECT version, num_records, num_spans, total_bytes FROM \"\".system.protected_ts_meta UNION ALL SELECT 0 AS version, 0 AS num_records, 0 AS num_spans, 0 AS total_bytes ORDER BY version DESC LIMIT 1), checks AS (SELECT new_version, new_num_records, new_num_spans, new_total_bytes, (((($1 > 0) AND (new_num_spans > $1)) OR (($2 > 0) AND (new_total_bytes > $2))) OR EXISTS (SELECT * FROM \"\".system.protected_ts_records WHERE id = $4)) AS failed FROM (SELECT version + 1 AS new_version, num_records + 1 AS new_num_records, num_spans + $3 AS new_num_spans, ((total_bytes + length($9)) + length($6)) + COALESCE(length($7), 0) AS new_total_bytes FROM \"\".\"\".current_meta)), updated_meta AS (UPSERT INTO \"\".system.protected_ts_meta(version, num_records, num_spans, total_bytes) (SELECT new_version, new_num_records, new_num_spans, new_total_bytes FROM \"\".\"\".checks WHERE NOT failed) RETURNING version, num_records, num_spans,
I210407 05:37:01.213629 24056 9@util/log/event_log.go:32  [n1,client=127.0.0.1:43592,hostssl,user=root,intExec=protectedts-protect] 7874 | total_bytes), new_record AS (INSERT INTO \"\".system.protected_ts_records(id, ts, meta_type, meta, num_spans, spans) (SELECT $4, $5, $6, $7, $8, $9 WHERE NOT EXISTS (SELECT * FROM \"\".\"\".checks WHERE failed)) RETURNING id) SELECT failed, num_spans AS prev_spans, total_bytes AS prev_total_bytes, version AS prev_version FROM \"\".\"\".checks, \"\".\"\".current_meta","User":"node","ApplicationName":"$ internal-protectedts-protect","PlaceholderValues":["32768","1048576","2","'\\x35450b3dac6a42f8963845f6aeda13ed'","'1617773821189136706.0000000000'","'jobs'","'\\x363437383334353730353736393230353737'","2","'\\x3a0a1a03f671892203f6718a3a0a1a03f67c892203f67c8a'"],"ExecMode":"exec-internal","NumRows":1,"Age":7.385963,"FullTableScan":true}
I210407 05:37:01.221806 24083 9@util/log/event_log.go:32  [n1,client=127.0.0.1:43592,hostssl,user=root,intExec=log-job] 7875 ={"Timestamp":1617773821215382001,"EventType":"query_execute","Statement":"SELECT status, payload, progress, claim_session_id FROM \"\".system.jobs WHERE id = $1 FOR UPDATE","User":"root","ApplicationName":"$ internal-log-job","PlaceholderValues":["647834570576920577"],"ExecMode":"exec-internal","NumRows":1,"Age":6.382844}
I210407 05:37:01.223277 23960 9@util/log/event_log.go:32  [n1,client=127.0.0.1:43592,hostssl,user=root,intExec=job-update] 7876 ={"Timestamp":1617773821215382001,"EventType":"query_execute","Statement":"UPDATE \"\".system.jobs SET status = $2, payload = $3 WHERE id = $1","User":"root","ApplicationName":"$ internal-job-update","PlaceholderValues":["647834570576920577","'running'","'\\x0a364241434b555020444154414241534520736368656d616368616e676520544f20276e6f64656c6f63616c3a2f2f302f6261636b7570271204726f6f7418b1b8c38fb5ebef02321634434446474f535d5e5f60616465666e6f717778797c52360a00120a08c2c29e978197dfb9161a146e6f64656c6f63616c3a2f2f302f6261636b75703a1035450b3dac6a42f8963845f6aeda13ed'"],"ExecMode":"exec-internal","NumRows":1,"Age":1.300473}
I210407 05:37:01.224368 23982 jobs/registry.go:1175  [n1] 7877  BACKUP job 647834570576920577: stepping through state running with error: <nil>
I210407 05:37:01.228813 23983 9@util/log/event_log.go:32  [n1,job=647834570576920577,intExec=protectedts-GetRecord] 7878 ={"Timestamp":1617773821224460954,"EventType":"query_execute","Statement":"SELECT id, ts, meta_type, meta, spans, verified FROM \"\".system.protected_ts_records WHERE id = $1","User":"node","ApplicationName":"$ internal-protectedts-GetRecord","PlaceholderValues":["'\\x35450b3dac6a42f8963845f6aeda13ed'"],"ExecMode":"exec-internal","NumRows":1,"Age":4.321533}
I210407 05:37:01.230392 23965 9@util/log/event_log.go:32  [intExec=protectedts-GetMetadata] 7879 ={"Timestamp":1617773821229009942,"EventType":"query_execute","Statement":"WITH current_meta AS (SELECT version, num_records, num_spans, total_bytes FROM \"\".system.protected_ts_meta UNION ALL SELECT 0 AS version, 0 AS num_records, 0 AS num_spans, 0 AS total_bytes ORDER BY version DESC LIMIT 1) SELECT version, num_records, num_spans, total_bytes FROM \"\".\"\".current_meta","User":"node","ApplicationName":"$ internal-protectedts-GetMetadata","ExecMode":"exec-internal","NumRows":1,"Age":1.343026,"FullTableScan":true}
I210407 05:37:01.231921 23966 9@util/log/event_log.go:32  [intExec=protectedts-GetMetadata] 7880 ={"Timestamp":1617773821229009942,"EventType":"query_execute","Statement":"WITH current_meta AS (SELECT version, num_records, num_spans, total_bytes FROM \"\".system.protected_ts_meta UNION ALL SELECT 0 AS version, 0 AS num_records, 0 AS num_spans, 0 AS total_bytes ORDER BY version DESC LIMIT 1) SELECT version, num_records, num_spans, total_bytes FROM \"\".\"\".current_meta","User":"node","ApplicationName":"$ internal-protectedts-GetMetadata","ExecMode":"exec-internal","NumRows":1,"Age":1.412622,"FullTableScan":true}
I210407 05:37:01.232962 23969 9@util/log/event_log.go:32  [intExec=protectedts-GetRecords] 7881 ={"Timestamp":1617773821229009942,"EventType":"query_execute","Statement":"SELECT id, ts, meta_type, meta, spans, verified FROM \"\".system.protected_ts_records","User":"node","ApplicationName":"$ internal-protectedts-GetRecords","ExecMode":"exec-internal","NumRows":1,"Age":0.939326,"FullTableScan":true}
I210407 05:37:01.234726 24098 9@util/log/event_log.go:32  [n1,job=647834570576920577,intExec=protectedts-MarkVerified] 7882 ={"Timestamp":1617773821233092473,"EventType":"query_execute","Statement":"UPDATE \"\".system.protected_ts_records SET verified = true WHERE id = $1 RETURNING true","User":"node","ApplicationName":"$ internal-protectedts-MarkVerified","PlaceholderValues":["'\\x35450b3dac6a42f8963845f6aeda13ed'"],"ExecMode":"exec-internal","NumRows":1,"Age":1.607529}
I210407 05:37:01.245641 24110 ccl/backupccl/backup_processor.go:309  [n1,job=647834570576920577] 7883  sending ExportRequest for span /Table/113/{1-2} (attempt 1, priority normal)
I210407 05:37:01.245899 24110 ccl/storageccl/export.go:98  [n1,s1,r470/1:/{Table/113-Max}] 7884  export [/Table/113/1,/Table/113/2)
I210407 05:37:01.245965 24106 ccl/backupccl/backup_processor.go:309  [n1,job=647834570576920577] 7885  sending ExportRequest for span /Table/124/{1-2} (attempt 1, priority normal)
I210407 05:37:01.246637 24106 ccl/storageccl/export.go:98  [n1,s1,r470/1:/{Table/113-Max}] 7886  export [/Table/124/1,/Table/124/2)
I210407 05:37:01.249985 24085 9@util/log/event_log.go:32  [n1,job=647834570576920577,intExec=log-job] 7887 ={"Timestamp":1617773821247227260,"EventType":"query_execute","Statement":"SELECT status, payload, progress, claim_session_id FROM \"\".system.jobs WHERE id = $1 FOR UPDATE","User":"root","ApplicationName":"$ internal-log-job","PlaceholderValues":["647834570576920577"],"ExecMode":"exec-internal","NumRows":1,"Age":2.671564}
I210407 05:37:01.251992 24087 9@util/log/event_log.go:32  [n1,job=647834570576920577,intExec=job-update] 7888 ={"Timestamp":1617773821247227260,"EventType":"query_execute","Statement":"UPDATE \"\".system.jobs SET progress = $2 WHERE id = $1","User":"root","ApplicationName":"$ internal-job-update","PlaceholderValues":["647834570576920577","'\\x0d0000003f10fbfdc48fb5ebef025200'"],"ExecMode":"exec-internal","NumRows":1,"Age":1.837092}
I210407 05:37:01.257529 24111 9@util/log/event_log.go:32  [n1,job=647834570576920577,intExec=log-job] 7889 ={"Timestamp":1617773821256017773,"EventType":"query_execute","Statement":"SELECT status, payload, progress, claim_session_id FROM \"\".system.jobs WHERE id = $1 FOR UPDATE","User":"root","ApplicationName":"$ internal-log-job","PlaceholderValues":["647834570576920577"],"ExecMode":"exec-internal","NumRows":1,"Age":1.46976}
I210407 05:37:01.259263 24113 9@util/log/event_log.go:32  [n1,job=647834570576920577,intExec=job-update] 7890 ={"Timestamp":1617773821256017773,"EventType":"query_execute","Statement":"UPDATE \"\".system.jobs SET progress = $2 WHERE id = $1","User":"root","ApplicationName":"$ internal-job-update","PlaceholderValues":["647834570576920577","'\\x0d0000803f10d2c2c58fb5ebef025200'"],"ExecMode":"exec-internal","NumRows":1,"Age":1.578996}
I210407 05:37:01.268333 24068 9@util/log/event_log.go:32  [n1,job=647834570576920577,intExec=get-table-statistics] 7891 ={"Timestamp":1617773821267834530,"EventType":"query_execute","Statement":"SELECT \"tableID\", \"statisticID\", name, \"columnIDs\", \"createdAt\", \"rowCount\", \"distinctCount\", \"nullCount\", histogram FROM \"\".system.table_statistics WHERE \"tableID\" = $1 ORDER BY \"createdAt\" DESC","User":"root","ApplicationName":"$ internal-get-table-statistics","PlaceholderValues":["124"],"ExecMode":"exec-internal","Age":1.46611,"TxnCounter":1}
I210407 05:37:01.275459 24069 9@util/log/event_log.go:32  [n1,job=647834570576920577,intExec=protectedts-Release] 7892 ={"Timestamp":1617773821270684773,"EventType":"query_execute","Statement":"WITH current_meta AS (SELECT version, num_records, num_spans, total_bytes FROM \"\".system.protected_ts_meta UNION ALL SELECT 0 AS version, 0 AS num_records, 0 AS num_spans, 0 AS total_bytes ORDER BY version DESC LIMIT 1), record AS (SELECT id, num_spans AS record_spans, (length(spans) + length(meta_type)) + COALESCE(length(meta), 0) AS record_bytes FROM \"\".system.protected_ts_records WHERE id = $1), updated_meta AS (UPSERT INTO \"\".system.protected_ts_meta(version, num_records, num_spans, total_bytes) (SELECT version, num_records, num_spans, total_bytes FROM (SELECT version + 1 AS version, num_records - 1 AS num_records, num_spans - record_spans AS num_spans, total_bytes - record_bytes AS total_bytes FROM \"\".\"\".current_meta RIGHT JOIN \"\".\"\".record ON true)) RETURNING 1) DELETE FROM \"\".system.protected_ts_records AS r WHERE EXISTS (SELECT NULL FROM \"\".\"\".record WHERE r.id = record.id) RETURNING NULL","User":"node","Appli
I210407 05:37:01.275459 24069 9@util/log/event_log.go:32  [n1,job=647834570576920577,intExec=protectedts-Release] 7892 |cationName":"$ internal-protectedts-Release","PlaceholderValues":["'\\x35450b3dac6a42f8963845f6aeda13ed'"],"ExecMode":"exec-internal","NumRows":1,"Age":4.726057,"FullTableScan":true}
I210407 05:37:01.279277 24132 9@util/log/event_log.go:32  [n1,job=647834570576920577,intExec=lookup-schedule-info] 7893 ={"Timestamp":1617773821277720416,"EventType":"query_execute","Statement":"SELECT created_by_id FROM \"\".system.jobs WHERE (id = $1) AND (created_by_type = $2)","User":"node","ApplicationName":"$ internal-lookup-schedule-info","PlaceholderValues":["647834570576920577","'crdb_schedule'"],"ExecMode":"exec-internal","Age":1.525738}
I210407 05:37:01.279398 23982 jobs/registry.go:1175  [n1] 7894  BACKUP job 647834570576920577: stepping through state succeeded with error: <nil>
I210407 05:37:01.281056 24133 9@util/log/event_log.go:32  [n1,intExec=log-job] 7895 ={"Timestamp":1617773821279460151,"EventType":"query_execute","Statement":"SELECT status, payload, progress, claim_session_id FROM \"\".system.jobs WHERE id = $1 FOR UPDATE","User":"root","ApplicationName":"$ internal-log-job","PlaceholderValues":["647834570576920577"],"ExecMode":"exec-internal","NumRows":1,"Age":1.568175}
I210407 05:37:01.282926 24135 9@util/log/event_log.go:32  [n1,intExec=job-update] 7896 ={"Timestamp":1617773821279460151,"EventType":"query_execute","Statement":"UPDATE \"\".system.jobs SET status = $2, payload = $3, progress = $4 WHERE id = $1","User":"root","ApplicationName":"$ internal-job-update","PlaceholderValues":["647834570576920577","'succeeded'","'\\x0a364241434b555020444154414241534520736368656d616368616e676520544f20276e6f64656c6f63616c3a2f2f302f6261636b7570271204726f6f7418b1b8c38fb5ebef02208287c78fb5ebef02321634434446474f535d5e5f60616465666e6f717778797c52360a00120a08c2c29e978197dfb9161a146e6f64656c6f63616c3a2f2f302f6261636b75703a1035450b3dac6a42f8963845f6aeda13ed'","'\\x0d0000803f10e4f9c68fb5ebef025200'"],"ExecMode":"exec-internal","NumRows":1,"Age":1.724564}
I210407 05:37:01.293740 4787 9@util/log/event_log.go:32  [n1,client=127.0.0.1:43592,hostssl,user=root] 7897 ={"Timestamp":1617773821188946861,"EventType":"query_execute","Statement":"BACKUP DATABASE schemachange TO 'nodelocal://0/backup'","User":"root","ExecMode":"exec","NumRows":1,"Age":104.82407,"TxnCounter":7}
    schema_change_external_test.go:81: backup in /go/src/github.com/cockroachdb/cockroach/artifacts/TestWorkload589379874
I210407 05:37:01.293956 8 testutils/testcluster/testcluster.go:101  [-] 7898  TestCluster quiescing nodes
E210407 05:37:01.294283 24137 kv/kvclient/kvcoord/txn_interceptor_committer.go:452  [-] 7899  making txn commit explicit failed for "unnamed" meta={id=c4f1e036 pri=0.01046769 epo=0 ts=1617773821.279460151,0 min=1617773821.279460151,0 seq=6} lock=true stat=STAGING rts=1617773821.279460151,0 wto=false gul=1617773821.779460151,0 int=1 ifw=5: result is ambiguous (server shutdown)
W210407 05:37:01.294441 383 jobs/registry.go:729  [-] 7900  canceling all adopted jobs due to stopper quiescing
W210407 05:37:01.294827 1806 jobs/registry.go:729  [-] 7901  canceling all adopted jobs due to stopper quiescing
W210407 05:37:01.295372 1805 sql/sqlliveness/slinstance/slinstance.go:183  [n3] 7902  exiting heartbeat loop
W210407 05:37:01.295977 2114 kv/kvserver/closedts/sidetransport/receiver.go:125  [n3] 7903  closed timestamps side-transport connection dropped from node: 1
W210407 05:37:01.296327 1191 2@gossip/gossip.go:1491  [n2] 7904  no incoming or outgoing connections
W210407 05:37:01.296497 1398 jobs/registry.go:729  [-] 7905  canceling all adopted jobs due to stopper quiescing
W210407 05:37:01.296743 2111 kv/kvserver/closedts/sidetransport/receiver.go:125  [n2] 7906  closed timestamps side-transport connection dropped from node: 1
W210407 05:37:01.296761 1397 sql/sqlliveness/slinstance/slinstance.go:183  [n2] 7907  exiting heartbeat loop
W210407 05:37:01.297196 382 sql/sqlliveness/slinstance/slinstance.go:183  [n1] 7908  exiting heartbeat loop
--- FAIL: TestWorkload (14.16s)

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=TestWorkload PKG=./pkg/ccl/testccl/workload/schemachange TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestWorkload.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[ccl/testccl/workload/schemachange.TestWorkload failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2859990&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=2859990&tab=artifacts#/) on [master @ 704a9fa4ffd144a2bc977bc1a6853a853b17e70e](https://github.com/cockroachdb/cockroach/commits/704a9fa4ffd144a2bc977bc1a6853a853b17e70e):


```
I210408 05:35:29.769561 19838 9@util/log/event_log.go:32  [intExec=protectedts-GetMetadata] 6122 ={"Timestamp":1617860129767074827,"EventType":"query_execute","Statement":"WITH current_meta AS (SELECT version, num_records, num_spans, total_bytes FROM \"\".system.protected_ts_meta UNION ALL SELECT 0 AS version, 0 AS num_records, 0 AS num_spans, 0 AS total_bytes ORDER BY version DESC LIMIT 1) SELECT version, num_records, num_spans, total_bytes FROM \"\".\"\".current_meta","User":"node","ApplicationName":"$ internal-protectedts-GetMetadata","ExecMode":"exec-internal","NumRows":1,"Age":2.445782,"FullTableScan":true}
I210408 05:35:29.771362 19890 9@util/log/event_log.go:32  [intExec=protectedts-GetMetadata] 6123 ={"Timestamp":1617860129767074827,"EventType":"query_execute","Statement":"WITH current_meta AS (SELECT version, num_records, num_spans, total_bytes FROM \"\".system.protected_ts_meta UNION ALL SELECT 0 AS version, 0 AS num_records, 0 AS num_spans, 0 AS total_bytes ORDER BY version DESC LIMIT 1) SELECT version, num_records, num_spans, total_bytes FROM \"\".\"\".current_meta","User":"node","ApplicationName":"$ internal-protectedts-GetMetadata","ExecMode":"exec-internal","NumRows":1,"Age":1.647199,"FullTableScan":true}
I210408 05:35:29.773016 19894 9@util/log/event_log.go:32  [intExec=protectedts-GetRecords] 6124 ={"Timestamp":1617860129767074827,"EventType":"query_execute","Statement":"SELECT id, ts, meta_type, meta, spans, verified FROM \"\".system.protected_ts_records","User":"node","ApplicationName":"$ internal-protectedts-GetRecords","ExecMode":"exec-internal","NumRows":1,"Age":1.564139,"FullTableScan":true}
I210408 05:35:29.776503 19906 9@util/log/event_log.go:32  [intExec=protectedts-GetMetadata] 6125 ={"Timestamp":1617860129773503612,"EventType":"query_execute","Statement":"WITH current_meta AS (SELECT version, num_records, num_spans, total_bytes FROM \"\".system.protected_ts_meta UNION ALL SELECT 0 AS version, 0 AS num_records, 0 AS num_spans, 0 AS total_bytes ORDER BY version DESC LIMIT 1) SELECT version, num_records, num_spans, total_bytes FROM \"\".\"\".current_meta","User":"node","ApplicationName":"$ internal-protectedts-GetMetadata","ExecMode":"exec-internal","NumRows":1,"Age":2.959919,"FullTableScan":true}
I210408 05:35:29.778575 19907 9@util/log/event_log.go:32  [intExec=protectedts-GetMetadata] 6126 ={"Timestamp":1617860129773503612,"EventType":"query_execute","Statement":"WITH current_meta AS (SELECT version, num_records, num_spans, total_bytes FROM \"\".system.protected_ts_meta UNION ALL SELECT 0 AS version, 0 AS num_records, 0 AS num_spans, 0 AS total_bytes ORDER BY version DESC LIMIT 1) SELECT version, num_records, num_spans, total_bytes FROM \"\".\"\".current_meta","User":"node","ApplicationName":"$ internal-protectedts-GetMetadata","ExecMode":"exec-internal","NumRows":1,"Age":1.958022,"FullTableScan":true}
I210408 05:35:29.780742 19902 9@util/log/event_log.go:32  [intExec=protectedts-GetRecords] 6127 ={"Timestamp":1617860129773503612,"EventType":"query_execute","Statement":"SELECT id, ts, meta_type, meta, spans, verified FROM \"\".system.protected_ts_records","User":"node","ApplicationName":"$ internal-protectedts-GetRecords","ExecMode":"exec-internal","NumRows":1,"Age":2.054503,"FullTableScan":true}
I210408 05:35:29.783984 19904 9@util/log/event_log.go:32  [n1,job=648117386407346177,intExec=protectedts-MarkVerified] 6128 ={"Timestamp":1617860129782178459,"EventType":"query_execute","Statement":"UPDATE \"\".system.protected_ts_records SET verified = true WHERE id = $1 RETURNING true","User":"node","ApplicationName":"$ internal-protectedts-MarkVerified","PlaceholderValues":["'\\xb0932e071eb44f89ae1215b3fca07504'"],"ExecMode":"exec-internal","NumRows":1,"Age":1.75478}
I210408 05:35:29.793940 19941 ccl/backupccl/backup_processor.go:309  [n2] 6129  sending ExportRequest for span /Table/66/1{-0} (attempt 1, priority normal)
I210408 05:35:29.793972 19886 ccl/backupccl/backup_processor.go:309  [n3] 6130  sending ExportRequest for span /Table/83/{1-2} (attempt 1, priority normal)
I210408 05:35:29.794183 19941 ccl/storageccl/export.go:98  [n2,s2,r48/2:/Table/{66-71}] 6131  export [/Table/66/1,/Table/66/10)
I210408 05:35:29.794229 19887 ccl/backupccl/backup_processor.go:309  [n3] 6133  sending ExportRequest for span /Table/87/{1-2} (attempt 1, priority normal)
I210408 05:35:29.794386 19887 ccl/storageccl/export.go:98  [n3,s3,r360/3:/{Table/83-Max}] 6134  export [/Table/87/1,/Table/87/2)
I210408 05:35:29.793944 19955 ccl/backupccl/backup_processor.go:309  [n3] 6135  sending ExportRequest for span /Table/71/{1-2} (attempt 1, priority normal)
I210408 05:35:29.794930 19955 ccl/storageccl/export.go:98  [n3,s3,r58/3:/Table/7{1-2}] 6136  export [/Table/71/1,/Table/71/2)
I210408 05:35:29.794956 19889 ccl/backupccl/backup_processor.go:309  [n3] 6137  sending ExportRequest for span /Table/88/{1-7} (attempt 1, priority normal)
I210408 05:35:29.794192 19886 ccl/storageccl/export.go:98  [n3,s3,r360/3:/{Table/83-Max}] 6132  export [/Table/83/1,/Table/83/2)
I210408 05:35:29.795310 19889 ccl/storageccl/export.go:98  [n3,s3,r360/3:/{Table/83-Max}] 6138  export [/Table/88/1,/Table/88/7)
I210408 05:35:29.798181 19786 9@util/log/event_log.go:32  [n1,job=648117386407346177,intExec=log-job] 6139 ={"Timestamp":1617860129795818511,"EventType":"query_execute","Statement":"SELECT status, payload, progress, claim_session_id FROM \"\".system.jobs WHERE id = $1 FOR UPDATE","User":"root","ApplicationName":"$ internal-log-job","PlaceholderValues":["648117386407346177"],"ExecMode":"exec-internal","NumRows":1,"Age":2.319869}
I210408 05:35:29.799948 19788 9@util/log/event_log.go:32  [n1,job=648117386407346177,intExec=job-update] 6140 ={"Timestamp":1617860129795818511,"EventType":"query_execute","Statement":"UPDATE \"\".system.jobs SET progress = $2 WHERE id = $1","User":"root","ApplicationName":"$ internal-job-update","PlaceholderValues":["648117386407346177","'\\x0dcdcc4c3e10ebddd4d2f6edef025200'"],"ExecMode":"exec-internal","NumRows":1,"Age":1.623773}
I210408 05:35:29.803767 19944 9@util/log/event_log.go:32  [n1,job=648117386407346177,intExec=log-job] 6141 ={"Timestamp":1617860129802273798,"EventType":"query_execute","Statement":"SELECT status, payload, progress, claim_session_id FROM \"\".system.jobs WHERE id = $1 FOR UPDATE","User":"root","ApplicationName":"$ internal-log-job","PlaceholderValues":["648117386407346177"],"ExecMode":"exec-internal","NumRows":1,"Age":1.462821}
I210408 05:35:29.805444 19862 9@util/log/event_log.go:32  [n2,intExec=stmt-diag-poll] 6143 ={"Timestamp":1617860129803797507,"EventType":"query_execute","Statement":"SELECT id, statement_fingerprint FROM \"\".system.statement_diagnostics_requests WHERE completed = false","User":"root","ApplicationName":"$ internal-stmt-diag-poll","ExecMode":"exec-internal","Age":1.860082,"TxnCounter":1}
I210408 05:35:29.805435 19946 9@util/log/event_log.go:32  [n1,job=648117386407346177,intExec=job-update] 6142 ={"Timestamp":1617860129802273798,"EventType":"query_execute","Statement":"UPDATE \"\".system.jobs SET progress = $2 WHERE id = $1","User":"root","ApplicationName":"$ internal-job-update","PlaceholderValues":["648117386407346177","'\\x0d0000803f10a290d5d2f6edef025200'"],"ExecMode":"exec-internal","NumRows":1,"Age":1.543674}
I210408 05:35:29.808100 1725 2@server/status/runtime.go:553  [n3] 6144  runtime stats: 271 MiB RSS, 868 goroutines (stacks: 12 MiB), 104 MiB/161 MiB Go alloc/total (heap fragmentation: 19 MiB, heap reserved: 11 MiB, heap released: 47 MiB), 29 MiB/38 MiB CGO alloc/total (0.0 CGO/sec), 0.0/0.0 %(u/s)time, 0.0 %gc (0x), 26 MiB/26 MiB (r/w)net
I210408 05:35:29.813196 19806 9@util/log/event_log.go:32  [n2,intExec=update-session] 6145 ={"Timestamp":1617860129810358544,"EventType":"query_execute","Statement":"UPDATE \"\".\"\".sqlliveness SET expiration = $1 WHERE session_id = $2 RETURNING session_id","User":"node","ApplicationName":"$ internal-update-session","PlaceholderValues":["1617860169810356069.0000000000","'\\xe712cfa5e2f34fa5af6f2d8b59a3a5a1'"],"ExecMode":"exec-internal","NumRows":1,"Age":2.775207}
I210408 05:35:29.820332 19949 9@util/log/event_log.go:32  [n1,job=648117386407346177,intExec=get-table-statistics] 6146 ={"Timestamp":1617860129819427511,"EventType":"query_execute","Statement":"SELECT \"tableID\", \"statisticID\", name, \"columnIDs\", \"createdAt\", \"rowCount\", \"distinctCount\", \"nullCount\", histogram FROM \"\".system.table_statistics WHERE \"tableID\" = $1 ORDER BY \"createdAt\" DESC","User":"root","ApplicationName":"$ internal-get-table-statistics","PlaceholderValues":["83"],"ExecMode":"exec-internal","Age":2.099166,"TxnCounter":1}
I210408 05:35:29.823995 19950 9@util/log/event_log.go:32  [n1,job=648117386407346177,intExec=get-table-statistics] 6147 ={"Timestamp":1617860129822779596,"EventType":"query_execute","Statement":"SELECT \"tableID\", \"statisticID\", name, \"columnIDs\", \"createdAt\", \"rowCount\", \"distinctCount\", \"nullCount\", histogram FROM \"\".system.table_statistics WHERE \"tableID\" = $1 ORDER BY \"createdAt\" DESC","User":"root","ApplicationName":"$ internal-get-table-statistics","PlaceholderValues":["87"],"ExecMode":"exec-internal","Age":2.828108,"TxnCounter":1}
I210408 05:35:29.832930 20038 9@util/log/event_log.go:32  [intExec=fetch-single-session] 6148 ={"Timestamp":1617860129830223231,"EventType":"query_execute","Statement":"SELECT expiration FROM \"\".\"\".sqlliveness WHERE session_id = $1","User":"node","ApplicationName":"$ internal-fetch-single-session","PlaceholderValues":["'\\xf6325aa92b694e77b1e8bb2c89cfcebe'"],"ExecMode":"exec-internal","NumRows":1,"Age":2.660213}
I210408 05:35:29.833327 20003 9@util/log/event_log.go:32  [intExec=expire-sessions] 6149 ={"Timestamp":1617860129825259894,"EventType":"query_execute","Statement":"UPDATE \"\".system.jobs SET claim_session_id = NULL WHERE ((claim_session_id != $1) AND (status IN ('running', 'pending', 'cancel-requested', 'pause-requested', 'reverting'))) AND (NOT crdb_internal.sql_liveness_is_alive(claim_session_id))","User":"root","ApplicationName":"$ internal-expire-sessions","PlaceholderValues":["'\\xe712cfa5e2f34fa5af6f2d8b59a3a5a1'"],"ExecMode":"exec-internal","Age":11.741849,"TxnCounter":1}
I210408 05:35:29.834827 20006 9@util/log/event_log.go:32  [n1,job=648117386407346177,intExec=protectedts-Release] 6150 ={"Timestamp":1617860129827348501,"EventType":"query_execute","Statement":"WITH current_meta AS (SELECT version, num_records, num_spans, total_bytes FROM \"\".system.protected_ts_meta UNION ALL SELECT 0 AS version, 0 AS num_records, 0 AS num_spans, 0 AS total_bytes ORDER BY version DESC LIMIT 1), record AS (SELECT id, num_spans AS record_spans, (length(spans) + length(meta_type)) + COALESCE(length(meta), 0) AS record_bytes FROM \"\".system.protected_ts_records WHERE id = $1), updated_meta AS (UPSERT INTO \"\".system.protected_ts_meta(version, num_records, num_spans, total_bytes) (SELECT version, num_records, num_spans, total_bytes FROM (SELECT version + 1 AS version, num_records - 1 AS num_records, num_spans - record_spans AS num_spans, total_bytes - record_bytes AS total_bytes FROM \"\".\"\".current_meta RIGHT JOIN \"\".\"\".record ON true)) RETURNING 1) DELETE FROM \"\".system.protected_ts_records AS r WHERE EXISTS (SELECT NULL FROM \"\".\"\".record WHERE r.id = record.id) RETURNING NULL","User":"node","Appli
I210408 05:35:29.834827 20006 9@util/log/event_log.go:32  [n1,job=648117386407346177,intExec=protectedts-Release] 6150 |cationName":"$ internal-protectedts-Release","PlaceholderValues":["'\\xb0932e071eb44f89ae1215b3fca07504'"],"ExecMode":"exec-internal","NumRows":1,"Age":7.409777,"FullTableScan":true}
I210408 05:35:29.839328 20042 9@util/log/event_log.go:32  [intExec=cancel/pause-requested] 6151 ={"Timestamp":1617860129833453313,"EventType":"query_execute","Statement":"UPDATE \"\".system.jobs SET status = CASE WHEN status = $1 THEN $2 WHEN status = $3 THEN $4 ELSE status END WHERE (status IN ($1, $3)) AND ((claim_session_id = $5) AND (claim_instance_id = $6)) RETURNING id, status","User":"node","ApplicationName":"$ internal-cancel/pause-requested","PlaceholderValues":["'pause-requested'","'paused'","'cancel-requested'","'reverting'","'\\xe712cfa5e2f34fa5af6f2d8b59a3a5a1'","2"],"ExecMode":"exec-internal","Age":5.819091}
I210408 05:35:29.854370 20026 9@util/log/event_log.go:32  [n1,job=648117386407346177,intExec=lookup-schedule-info] 6152 ={"Timestamp":1617860129852964695,"EventType":"query_execute","Statement":"SELECT created_by_id FROM \"\".system.jobs WHERE (id = $1) AND (created_by_type = $2)","User":"node","ApplicationName":"$ internal-lookup-schedule-info","PlaceholderValues":["648117386407346177","'crdb_schedule'"],"ExecMode":"exec-internal","Age":1.374638}
I210408 05:35:29.854546 19830 jobs/registry.go:1175  [n1] 6153  BACKUP job 648117386407346177: stepping through state succeeded with error: <nil>
I210408 05:35:29.856886 20027 9@util/log/event_log.go:32  [n1,intExec=log-job] 6154 ={"Timestamp":1617860129854607000,"EventType":"query_execute","Statement":"SELECT status, payload, progress, claim_session_id FROM \"\".system.jobs WHERE id = $1 FOR UPDATE","User":"root","ApplicationName":"$ internal-log-job","PlaceholderValues":["648117386407346177"],"ExecMode":"exec-internal","NumRows":1,"Age":2.248649}
I210408 05:35:29.859287 19981 9@util/log/event_log.go:32  [n1,intExec=job-update] 6155 ={"Timestamp":1617860129854607000,"EventType":"query_execute","Statement":"UPDATE \"\".system.jobs SET status = $2, payload = $3, progress = $4 WHERE id = $1","User":"root","ApplicationName":"$ internal-job-update","PlaceholderValues":["648117386407346177","'succeeded'","'\\x0a364241434b555020444154414241534520736368656d616368616e676520544f20276e6f64656c6f63616c3a2f2f302f6261636b7570271204726f6f741889cbd2d2f6edef0220e4bbd8d2f6edef023211343536393a3b4247494a4b4e4f5357585f52360a00120a089bf9e9c7f5e6f2b9161a146e6f64656c6f63616c3a2f2f302f6261636b75703a10b0932e071eb44f89ae1215b3fca07504'","'\\x0d0000803f108fa9d8d2f6edef025200'"],"ExecMode":"exec-internal","NumRows":1,"Age":2.246041}
I210408 05:35:29.860959 4770 9@util/log/event_log.go:32  [n1,client=127.0.0.1:55468,hostssl,user=root] 6156 ={"Timestamp":1617860129732426970,"EventType":"query_execute","Statement":"BACKUP DATABASE schemachange TO 'nodelocal://0/backup'","User":"root","ExecMode":"exec","NumRows":1,"Age":128.57576,"TxnCounter":7}
    schema_change_external_test.go:81: backup in /go/src/github.com/cockroachdb/cockroach/artifacts/TestWorkload328213678
I210408 05:35:29.861650 55 testutils/testcluster/testcluster.go:101  [-] 6157  TestCluster quiescing nodes
W210408 05:35:29.862153 1919 jobs/registry.go:729  [-] 6158  canceling all adopted jobs due to stopper quiescing
W210408 05:35:29.862404 1918 sql/sqlliveness/slinstance/slinstance.go:183  [n3] 6159  exiting heartbeat loop
W210408 05:35:29.862719 386 jobs/registry.go:729  [-] 6160  canceling all adopted jobs due to stopper quiescing
W210408 05:35:29.862940 6593 kv/kvserver/closedts/sidetransport/receiver.go:125  [n3] 6161  closed timestamps side-transport connection dropped from node: 2
W210408 05:35:29.863018 2519 kv/kvserver/closedts/sidetransport/receiver.go:125  [n3] 6162  closed timestamps side-transport connection dropped from node: 1
W210408 05:35:29.863502 6393 kv/kvserver/closedts/sidetransport/receiver.go:125  [n1] 6163  closed timestamps side-transport connection dropped from node: 2
W210408 05:35:29.863584 1515 sql/sqlliveness/slinstance/slinstance.go:183  [n2] 6164  exiting heartbeat loop
W210408 05:35:29.863797 385 sql/sqlliveness/slinstance/slinstance.go:183  [n1] 6165  exiting heartbeat loop
W210408 05:35:29.863840 1804 kv/kvserver/raft_transport.go:633  [n2] 6166  while processing outgoing Raft queue to node 1: EOF:
W210408 05:35:29.863897 1516 jobs/registry.go:729  [-] 6167  canceling all adopted jobs due to stopper quiescing
W210408 05:35:29.863919 1869 kv/kvserver/raft_transport.go:633  [n2] 6168  while processing outgoing Raft queue to node 1: EOF:
--- FAIL: TestWorkload (10.37s)
```
<details><summary>Reproduce</summary>
<p>
<p>To reproduce, try:

```bash
make stressrace TESTS=TestWorkload PKG=./pkg/ccl/testccl/workload/schemachange TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
</p>
<p>Parameters in this failure:

- GOFLAGS=-parallel=4
</p>
</p>
</details>

[See this test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestWorkload.*&sort=title+created&display=lastcommented+project)

<sub>[Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub>

[ccl/testccl/workload/schemachange.TestWorkload failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2863454&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=2863454&tab=artifacts#/) on [master @ c41bd9cf5aa3997187310c18773ca787a88ea5db](https://github.com/cockroachdb/cockroach/commits/c41bd9cf5aa3997187310c18773ca787a88ea5db):


```
52, 71, _enum102, 78
52, 71, _enum77, 74
52, 71, enum102, 77
52, 71, enum77, 73
I210408 21:12:13.615515 21833 9@util/log/event_log.go:32  [n1,client=127.0.0.1:51986,hostssl,user=root,intExec=backup-lookup-tenant] 5276 ={"Timestamp":1617916333592426930,"EventType":"query_execute","Statement":"SELECT id, active, info FROM \"\".system.tenants","User":"root","ApplicationName":"$ internal-backup-lookup-tenant","ExecMode":"exec-internal","Age":1.755126,"FullTableScan":true}
I210408 21:12:13.626845 21834 9@util/log/event_log.go:32  [n1,client=127.0.0.1:51986,hostssl,user=root,intExec=job-row-insert] 5277 ={"Timestamp":1617916333592426930,"EventType":"query_execute","Statement":"INSERT INTO \"\".system.jobs(id, status, payload, progress, claim_session_id, claim_instance_id) VALUES ($1, $2, $3, $4, $5, $6)","User":"root","ApplicationName":"$ internal-job-row-insert","PlaceholderValues":["648301555277627393","'running'","'\\x0a364241434b555020444154414241534520736368656d616368616e676520544f20276e6f64656c6f63616c3a2f2f302f6261636b7570271204726f6f74320c343c3d3e3f404748494a4d4e52360a00120a08c2d896a1d5caffb9161a146e6f64656c6f63616c3a2f2f302f6261636b75703a10132bf83af180480985f2becfe2f61478'","'\\x10eb9edc82c8efef025200'","'\\x3b24a8a6d4604fdbb817c459676835e5'","1"],"ExecMode":"exec-internal","NumRows":1,"Age":9.278057}
I210408 21:12:13.644091 21836 9@util/log/event_log.go:32  [n1,client=127.0.0.1:51986,hostssl,user=root,intExec=protectedts-protect] 5278 ={"Timestamp":1617916333592426930,"EventType":"query_execute","Statement":"WITH current_meta AS (SELECT version, num_records, num_spans, total_bytes FROM \"\".system.protected_ts_meta UNION ALL SELECT 0 AS version, 0 AS num_records, 0 AS num_spans, 0 AS total_bytes ORDER BY version DESC LIMIT 1), checks AS (SELECT new_version, new_num_records, new_num_spans, new_total_bytes, (((($1 > 0) AND (new_num_spans > $1)) OR (($2 > 0) AND (new_total_bytes > $2))) OR EXISTS (SELECT * FROM \"\".system.protected_ts_records WHERE id = $4)) AS failed FROM (SELECT version + 1 AS new_version, num_records + 1 AS new_num_records, num_spans + $3 AS new_num_spans, ((total_bytes + length($9)) + length($6)) + COALESCE(length($7), 0) AS new_total_bytes FROM \"\".\"\".current_meta)), updated_meta AS (UPSERT INTO \"\".system.protected_ts_meta(version, num_records, num_spans, total_bytes) (SELECT new_version, new_num_records, new_num_spans, new_total_bytes FROM \"\".\"\".checks WHERE NOT failed) RETURNING version, num_records, num_spans,
I210408 21:12:13.644091 21836 9@util/log/event_log.go:32  [n1,client=127.0.0.1:51986,hostssl,user=root,intExec=protectedts-protect] 5278 | total_bytes), new_record AS (INSERT INTO \"\".system.protected_ts_records(id, ts, meta_type, meta, num_spans, spans) (SELECT $4, $5, $6, $7, $8, $9 WHERE NOT EXISTS (SELECT * FROM \"\".\"\".checks WHERE failed)) RETURNING id) SELECT failed, num_spans AS prev_spans, total_bytes AS prev_total_bytes, version AS prev_version FROM \"\".\"\".checks, \"\".\"\".current_meta","User":"node","ApplicationName":"$ internal-protectedts-protect","PlaceholderValues":["32768","1048576","2","'\\x132bf83af180480985f2becfe2f61478'","'1617916333593570370.0000000000'","'jobs'","'\\x363438333031353535323737363237333933'","2","'\\x3a081a02c8892202c88b3a081a02d0892202d093'"],"ExecMode":"exec-internal","NumRows":1,"Age":9.454599,"FullTableScan":true}
I210408 21:12:13.650498 21838 9@util/log/event_log.go:32  [n1,client=127.0.0.1:51986,hostssl,user=root,intExec=log-job] 5279 ={"Timestamp":1617916333647210635,"EventType":"query_execute","Statement":"SELECT status, payload, progress, claim_session_id FROM \"\".system.jobs WHERE id = $1 FOR UPDATE","User":"root","ApplicationName":"$ internal-log-job","PlaceholderValues":["648301555277627393"],"ExecMode":"exec-internal","NumRows":1,"Age":3.220128}
I210408 21:12:13.672692 21883 9@util/log/event_log.go:32  [n1,client=127.0.0.1:51986,hostssl,user=root,intExec=job-update] 5280 ={"Timestamp":1617916333647210635,"EventType":"query_execute","Statement":"UPDATE \"\".system.jobs SET status = $2, payload = $3 WHERE id = $1","User":"root","ApplicationName":"$ internal-job-update","PlaceholderValues":["648301555277627393","'running'","'\\x0a364241434b555020444154414241534520736368656d616368616e676520544f20276e6f64656c6f63616c3a2f2f302f6261636b7570271204726f6f7418a8e6df82c8efef02320c343c3d3e3f404748494a4d4e52360a00120a08c2d896a1d5caffb9161a146e6f64656c6f63616c3a2f2f302f6261636b75703a10132bf83af180480985f2becfe2f61478'"],"ExecMode":"exec-internal","NumRows":1,"Age":21.868187}
I210408 21:12:13.675429 21423 jobs/registry.go:1175  [n1] 5281  BACKUP job 648301555277627393: stepping through state running with error: <nil>
I210408 21:12:13.678630 21424 9@util/log/event_log.go:32  [n1,job=648301555277627393,intExec=protectedts-GetRecord] 5282 ={"Timestamp":1617916333675591608,"EventType":"query_execute","Statement":"SELECT id, ts, meta_type, meta, spans, verified FROM \"\".system.protected_ts_records WHERE id = $1","User":"node","ApplicationName":"$ internal-protectedts-GetRecord","PlaceholderValues":["'\\x132bf83af180480985f2becfe2f61478'"],"ExecMode":"exec-internal","NumRows":1,"Age":2.98519}
I210408 21:12:13.683685 21887 9@util/log/event_log.go:32  [intExec=protectedts-GetMetadata] 5283 ={"Timestamp":1617916333679451408,"EventType":"query_execute","Statement":"WITH current_meta AS (SELECT version, num_records, num_spans, total_bytes FROM \"\".system.protected_ts_meta UNION ALL SELECT 0 AS version, 0 AS num_records, 0 AS num_spans, 0 AS total_bytes ORDER BY version DESC LIMIT 1) SELECT version, num_records, num_spans, total_bytes FROM \"\".\"\".current_meta","User":"node","ApplicationName":"$ internal-protectedts-GetMetadata","ExecMode":"exec-internal","NumRows":1,"Age":4.155276,"FullTableScan":true}
I210408 21:12:13.691226 21941 9@util/log/event_log.go:32  [intExec=protectedts-GetMetadata] 5284 ={"Timestamp":1617916333679451408,"EventType":"query_execute","Statement":"WITH current_meta AS (SELECT version, num_records, num_spans, total_bytes FROM \"\".system.protected_ts_meta UNION ALL SELECT 0 AS version, 0 AS num_records, 0 AS num_spans, 0 AS total_bytes ORDER BY version DESC LIMIT 1) SELECT version, num_records, num_spans, total_bytes FROM \"\".\"\".current_meta","User":"node","ApplicationName":"$ internal-protectedts-GetMetadata","ExecMode":"exec-internal","NumRows":1,"Age":7.305992,"FullTableScan":true}
I210408 21:12:13.694904 21944 9@util/log/event_log.go:32  [intExec=protectedts-GetRecords] 5285 ={"Timestamp":1617916333679451408,"EventType":"query_execute","Statement":"SELECT id, ts, meta_type, meta, spans, verified FROM \"\".system.protected_ts_records","User":"node","ApplicationName":"$ internal-protectedts-GetRecords","ExecMode":"exec-internal","NumRows":1,"Age":3.473901,"FullTableScan":true}
I210408 21:12:13.700129 21952 9@util/log/event_log.go:32  [intExec=protectedts-GetMetadata] 5286 ={"Timestamp":1617916333696188672,"EventType":"query_execute","Statement":"WITH current_meta AS (SELECT version, num_records, num_spans, total_bytes FROM \"\".system.protected_ts_meta UNION ALL SELECT 0 AS version, 0 AS num_records, 0 AS num_spans, 0 AS total_bytes ORDER BY version DESC LIMIT 1) SELECT version, num_records, num_spans, total_bytes FROM \"\".\"\".current_meta","User":"node","ApplicationName":"$ internal-protectedts-GetMetadata","ExecMode":"exec-internal","NumRows":1,"Age":3.878207,"FullTableScan":true}
I210408 21:12:13.705414 21954 9@util/log/event_log.go:32  [intExec=protectedts-GetMetadata] 5287 ={"Timestamp":1617916333696188672,"EventType":"query_execute","Statement":"WITH current_meta AS (SELECT version, num_records, num_spans, total_bytes FROM \"\".system.protected_ts_meta UNION ALL SELECT 0 AS version, 0 AS num_records, 0 AS num_spans, 0 AS total_bytes ORDER BY version DESC LIMIT 1) SELECT version, num_records, num_spans, total_bytes FROM \"\".\"\".current_meta","User":"node","ApplicationName":"$ internal-protectedts-GetMetadata","ExecMode":"exec-internal","NumRows":1,"Age":5.044866,"FullTableScan":true}
I210408 21:12:13.710614 21965 9@util/log/event_log.go:32  [intExec=protectedts-GetRecords] 5288 ={"Timestamp":1617916333696188672,"EventType":"query_execute","Statement":"SELECT id, ts, meta_type, meta, spans, verified FROM \"\".system.protected_ts_records","User":"node","ApplicationName":"$ internal-protectedts-GetRecords","ExecMode":"exec-internal","NumRows":1,"Age":4.942939,"FullTableScan":true}
I210408 21:12:13.756892 21972 9@util/log/event_log.go:32  [n1,job=648301555277627393,intExec=protectedts-MarkVerified] 5289 ={"Timestamp":1617916333749319336,"EventType":"query_execute","Statement":"UPDATE \"\".system.protected_ts_records SET verified = true WHERE id = $1 RETURNING true","User":"node","ApplicationName":"$ internal-protectedts-MarkVerified","PlaceholderValues":["'\\x132bf83af180480985f2becfe2f61478'"],"ExecMode":"exec-internal","NumRows":1,"Age":7.510892}
I210408 21:12:13.769586 21991 ccl/backupccl/backup_processor.go:309  [n2] 5290  sending ExportRequest for span /Table/72/1{-1} (attempt 1, priority normal)
I210408 21:12:13.769711 22002 ccl/backupccl/backup_processor.go:309  [n3] 5291  sending ExportRequest for span /Table/64/{1-3} (attempt 1, priority normal)
I210408 21:12:13.769950 21991 ccl/storageccl/export.go:98  [n2,s2,r49/2:/{Table/72-Max}] 5292  export [/Table/72/1,/Table/72/11)
I210408 21:12:13.770011 22002 ccl/storageccl/export.go:98  [n3,s3,r58/3:/Table/{64/2-70}] 5293  export [/Table/64/2,/Table/64/3)
I210408 21:12:13.770732 22003 ccl/storageccl/export.go:98  [n3,s3,r39/3:/Table/64{-/2}] 5294  export [/Table/64/1,/Table/64/2)
I210408 21:12:13.775256 21505 9@util/log/event_log.go:32  [n1,job=648301555277627393,intExec=log-job] 5295 ={"Timestamp":1617916333772179331,"EventType":"query_execute","Statement":"SELECT status, payload, progress, claim_session_id FROM \"\".system.jobs WHERE id = $1 FOR UPDATE","User":"root","ApplicationName":"$ internal-log-job","PlaceholderValues":["648301555277627393"],"ExecMode":"exec-internal","NumRows":1,"Age":2.984461}
I210408 21:12:13.778933 22019 9@util/log/event_log.go:32  [n1,job=648301555277627393,intExec=job-update] 5296 ={"Timestamp":1617916333772179331,"EventType":"query_execute","Statement":"UPDATE \"\".system.jobs SET progress = $2 WHERE id = $1","User":"root","ApplicationName":"$ internal-job-update","PlaceholderValues":["648301555277627393","'\\x0d0000003f10939be782c8efef025200'"],"ExecMode":"exec-internal","NumRows":1,"Age":3.445223}
I210408 21:12:13.794477 22022 9@util/log/event_log.go:32  [n1,job=648301555277627393,intExec=log-job] 5297 ={"Timestamp":1617916333789399411,"EventType":"query_execute","Statement":"SELECT status, payload, progress, claim_session_id FROM \"\".system.jobs WHERE id = $1 FOR UPDATE","User":"root","ApplicationName":"$ internal-log-job","PlaceholderValues":["648301555277627393"],"ExecMode":"exec-internal","NumRows":1,"Age":5.002898}
I210408 21:12:13.797350 22024 9@util/log/event_log.go:32  [n1,job=648301555277627393,intExec=job-update] 5298 ={"Timestamp":1617916333789399411,"EventType":"query_execute","Statement":"UPDATE \"\".system.jobs SET progress = $2 WHERE id = $1","User":"root","ApplicationName":"$ internal-job-update","PlaceholderValues":["648301555277627393","'\\x0d0000803f10d7a1e882c8efef025200'"],"ExecMode":"exec-internal","NumRows":1,"Age":2.658527}
I210408 21:12:13.828169 21591 9@util/log/event_log.go:32  [n1,job=648301555277627393,intExec=protectedts-Release] 5299 ={"Timestamp":1617916333811290373,"EventType":"query_execute","Statement":"WITH current_meta AS (SELECT version, num_records, num_spans, total_bytes FROM \"\".system.protected_ts_meta UNION ALL SELECT 0 AS version, 0 AS num_records, 0 AS num_spans, 0 AS total_bytes ORDER BY version DESC LIMIT 1), record AS (SELECT id, num_spans AS record_spans, (length(spans) + length(meta_type)) + COALESCE(length(meta), 0) AS record_bytes FROM \"\".system.protected_ts_records WHERE id = $1), updated_meta AS (UPSERT INTO \"\".system.protected_ts_meta(version, num_records, num_spans, total_bytes) (SELECT version, num_records, num_spans, total_bytes FROM (SELECT version + 1 AS version, num_records - 1 AS num_records, num_spans - record_spans AS num_spans, total_bytes - record_bytes AS total_bytes FROM \"\".\"\".current_meta RIGHT JOIN \"\".\"\".record ON true)) RETURNING 1) DELETE FROM \"\".system.protected_ts_records AS r WHERE EXISTS (SELECT NULL FROM \"\".\"\".record WHERE r.id = record.id) RETURNING NULL","User":"node","Appli
I210408 21:12:13.828169 21591 9@util/log/event_log.go:32  [n1,job=648301555277627393,intExec=protectedts-Release] 5299 |cationName":"$ internal-protectedts-Release","PlaceholderValues":["'\\x132bf83af180480985f2becfe2f61478'"],"ExecMode":"exec-internal","NumRows":1,"Age":16.788109,"FullTableScan":true}
I210408 21:12:13.845250 22009 9@util/log/event_log.go:32  [n1,job=648301555277627393,intExec=lookup-schedule-info] 5300 ={"Timestamp":1617916333842335948,"EventType":"query_execute","Statement":"SELECT created_by_id FROM \"\".system.jobs WHERE (id = $1) AND (created_by_type = $2)","User":"node","ApplicationName":"$ internal-lookup-schedule-info","PlaceholderValues":["648301555277627393","'crdb_schedule'"],"ExecMode":"exec-internal","Age":2.844365}
I210408 21:12:13.845693 21423 jobs/registry.go:1175  [n1] 5301  BACKUP job 648301555277627393: stepping through state succeeded with error: <nil>
I210408 21:12:13.848144 22010 9@util/log/event_log.go:32  [n1,intExec=log-job] 5302 ={"Timestamp":1617916333845886884,"EventType":"query_execute","Statement":"SELECT status, payload, progress, claim_session_id FROM \"\".system.jobs WHERE id = $1 FOR UPDATE","User":"root","ApplicationName":"$ internal-log-job","PlaceholderValues":["648301555277627393"],"ExecMode":"exec-internal","NumRows":1,"Age":2.203023}
I210408 21:12:13.850905 22012 9@util/log/event_log.go:32  [n1,intExec=job-update] 5303 ={"Timestamp":1617916333845886884,"EventType":"query_execute","Statement":"UPDATE \"\".system.jobs SET status = $2, payload = $3, progress = $4 WHERE id = $1","User":"root","ApplicationName":"$ internal-job-update","PlaceholderValues":["648301555277627393","'succeeded'","'\\x0a364241434b555020444154414241534520736368656d616368616e676520544f20276e6f64656c6f63616c3a2f2f302f6261636b7570271204726f6f7418a8e6df82c8efef0220f4eeeb82c8efef02320c343c3d3e3f404748494a4d4e52360a00120a08c2d896a1d5caffb9161a146e6f64656c6f63616c3a2f2f302f6261636b75703a10132bf83af180480985f2becfe2f61478'","'\\x0d0000803f10ffdaeb82c8efef025200'"],"ExecMode":"exec-internal","NumRows":1,"Age":2.408663}
I210408 21:12:13.853093 5426 9@util/log/event_log.go:32  [n1,client=127.0.0.1:51986,hostssl,user=root] 5304 ={"Timestamp":1617916333592426930,"EventType":"query_execute","Statement":"BACKUP DATABASE schemachange TO 'nodelocal://0/backup'","User":"root","ExecMode":"exec","NumRows":1,"Age":267.122,"TxnCounter":7}
    schema_change_external_test.go:81: backup in /go/src/github.com/cockroachdb/cockroach/artifacts/TestWorkload546534880
I210408 21:12:13.854054 12 testutils/testcluster/testcluster.go:101  [-] 5305  TestCluster quiescing nodes
W210408 21:12:13.854181 2067 sql/sqlliveness/slinstance/slinstance.go:183  [n3] 5306  exiting heartbeat loop
W210408 21:12:13.854340 495 sql/sqlliveness/slinstance/slinstance.go:183  [n1] 5307  exiting heartbeat loop
E210408 21:12:13.854565 22014 kv/kvclient/kvcoord/txn_interceptor_committer.go:452  [-] 5308  making txn commit explicit failed for "unnamed" meta={id=f9d81601 pri=0.00687968 epo=0 ts=1617916333.845886884,0 min=1617916333.845886884,0 seq=6} lock=true stat=STAGING rts=1617916333.845886884,0 wto=false gul=1617916334.345886884,0 int=1 ifw=5: result is ambiguous (server shutdown)
W210408 21:12:13.854765 2068 jobs/registry.go:729  [-] 5309  canceling all adopted jobs due to stopper quiescing
W210408 21:12:13.855035 7911 kv/kvserver/closedts/sidetransport/receiver.go:125  [n3] 5310  closed timestamps side-transport connection dropped from node: 2
W210408 21:12:13.855362 1658 jobs/registry.go:729  [-] 5311  canceling all adopted jobs due to stopper quiescing
W210408 21:12:13.855384 2554 kv/kvserver/closedts/sidetransport/receiver.go:125  [n2] 5312  closed timestamps side-transport connection dropped from node: 1
W210408 21:12:13.855201 10215 kv/kvserver/closedts/sidetransport/receiver.go:125  [n1] 5313  closed timestamps side-transport connection dropped from node: 3
W210408 21:12:13.856350 1657 sql/sqlliveness/slinstance/slinstance.go:183  [n2] 5314  exiting heartbeat loop
W210408 21:12:13.856488 7681 kv/kvserver/closedts/sidetransport/receiver.go:125  [n1] 5315  closed timestamps side-transport connection dropped from node: 2
W210408 21:12:13.856658 1369 2@gossip/gossip.go:1491  [n2] 5316  no incoming or outgoing connections
W210408 21:12:13.856687 496 jobs/registry.go:729  [-] 5317  canceling all adopted jobs due to stopper quiescing
--- FAIL: TestWorkload (19.87s)
```
<details><summary>Reproduce</summary>
<p>
<p>To reproduce, try:

```bash
make stressrace TESTS=TestWorkload PKG=./pkg/ccl/testccl/workload/schemachange TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
</p>
<p>Parameters in this failure:

- GOFLAGS=-json
</p>
</p>
</details>

[See this test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestWorkload.*&sort=title+created&display=lastcommented+project)

<sub>[Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub>

[ccl/testccl/workload/schemachange.TestWorkload failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2864514&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=2864514&tab=artifacts#/) on [master @ e1e10f99362e67771e00346fb0e006627eb67b09](https://github.com/cockroachdb/cockroach/commits/e1e10f99362e67771e00346fb0e006627eb67b09):


```
1, 29, ui, 14
1, 29, users, 4
1, 29, web_sessions, 19
1, 29, zones, 5
50, 0, public, 29
51, 0, public, 29
52, 0, public, 29
52, 29, _crdb_internal_region, 73
52, 29, _enum22, 64
52, 29, crdb_internal_region, 72
52, 29, enum22, 63
52, 29, table85, 75
I210409 04:39:41.085492 16777 9@util/log/event_log.go:32  [n1,client=127.0.0.1:38150,hostssl,user=root,intExec=backup-lookup-tenant] 4538 ={"Timestamp":1617943181075825091,"EventType":"query_execute","Statement":"SELECT id, active, info FROM \"\".system.tenants","User":"root","ApplicationName":"$ internal-backup-lookup-tenant","ExecMode":"exec-internal","Age":2.050445,"FullTableScan":true}
I210409 04:39:41.093716 16778 9@util/log/event_log.go:32  [n1,client=127.0.0.1:38150,hostssl,user=root,intExec=job-row-insert] 4539 ={"Timestamp":1617943181075825091,"EventType":"query_execute","Statement":"INSERT INTO \"\".system.jobs(id, status, payload, progress, claim_session_id, claim_instance_id) VALUES ($1, $2, $3, $4, $5, $6)","User":"root","ApplicationName":"$ internal-job-row-insert","PlaceholderValues":["648389529063161857","'running'","'\\x0a364241434b555020444154414241534520736368656d616368616e676520544f20276e6f64656c6f63616c3a2f2f302f6261636b7570271204726f6f743206343f4048494b52360a00120a08e5bf90f783d885ba161a146e6f64656c6f63616c3a2f2f302f6261636b75703a107458b94bb4b64d37b9378b4264feeded'","'\\x10f1cacc84acf0ef025200'","'\\x8629312e6a6d4f3c845c91e37af45ed0'","1"],"ExecMode":"exec-internal","NumRows":1,"Age":7.461581}
I210409 04:39:41.115641 16917 9@util/log/event_log.go:32  [n1,client=127.0.0.1:38150,hostssl,user=root,intExec=protectedts-protect] 4540 ={"Timestamp":1617943181075825091,"EventType":"query_execute","Statement":"WITH current_meta AS (SELECT version, num_records, num_spans, total_bytes FROM \"\".system.protected_ts_meta UNION ALL SELECT 0 AS version, 0 AS num_records, 0 AS num_spans, 0 AS total_bytes ORDER BY version DESC LIMIT 1), checks AS (SELECT new_version, new_num_records, new_num_spans, new_total_bytes, (((($1 > 0) AND (new_num_spans > $1)) OR (($2 > 0) AND (new_total_bytes > $2))) OR EXISTS (SELECT * FROM \"\".system.protected_ts_records WHERE id = $4)) AS failed FROM (SELECT version + 1 AS new_version, num_records + 1 AS new_num_records, num_spans + $3 AS new_num_spans, ((total_bytes + length($9)) + length($6)) + COALESCE(length($7), 0) AS new_total_bytes FROM \"\".\"\".current_meta)), updated_meta AS (UPSERT INTO \"\".system.protected_ts_meta(version, num_records, num_spans, total_bytes) (SELECT new_version, new_num_records, new_num_spans, new_total_bytes FROM \"\".\"\".checks WHERE NOT failed) RETURNING version, num_records, num_spans,
I210409 04:39:41.115641 16917 9@util/log/event_log.go:32  [n1,client=127.0.0.1:38150,hostssl,user=root,intExec=protectedts-protect] 4540 | total_bytes), new_record AS (INSERT INTO \"\".system.protected_ts_records(id, ts, meta_type, meta, num_spans, spans) (SELECT $4, $5, $6, $7, $8, $9 WHERE NOT EXISTS (SELECT * FROM \"\".\"\".checks WHERE failed)) RETURNING id) SELECT failed, num_spans AS prev_spans, total_bytes AS prev_total_bytes, version AS prev_version FROM \"\".\"\".checks, \"\".\"\".current_meta","User":"node","ApplicationName":"$ internal-protectedts-protect","PlaceholderValues":["32768","1048576","1","'\\x7458b94bb4b64d37b9378b4264feeded'","'1617943181077520357.0000000000'","'jobs'","'\\x363438333839353239303633313631383537'","1","'\\x3a081a02d3892202d38c'"],"ExecMode":"exec-internal","NumRows":1,"Age":8.879418,"FullTableScan":true}
I210409 04:39:41.131137 16870 9@util/log/event_log.go:32  [n1,client=127.0.0.1:38150,hostssl,user=root,intExec=log-job] 4541 ={"Timestamp":1617943181119124170,"EventType":"query_execute","Statement":"SELECT status, payload, progress, claim_session_id FROM \"\".system.jobs WHERE id = $1 FOR UPDATE","User":"root","ApplicationName":"$ internal-log-job","PlaceholderValues":["648389529063161857"],"ExecMode":"exec-internal","NumRows":1,"Age":11.937895}
I210409 04:39:41.150649 16872 9@util/log/event_log.go:32  [n1,client=127.0.0.1:38150,hostssl,user=root,intExec=job-update] 4542 ={"Timestamp":1617943181119124170,"EventType":"query_execute","Statement":"UPDATE \"\".system.jobs SET status = $2, payload = $3 WHERE id = $1","User":"root","ApplicationName":"$ internal-job-update","PlaceholderValues":["648389529063161857","'running'","'\\x0a364241434b555020444154414241534520736368656d616368616e676520544f20276e6f64656c6f63616c3a2f2f302f6261636b7570271204726f6f7418bb80d084acf0ef023206343f4048494b52360a00120a08e5bf90f783d885ba161a146e6f64656c6f63616c3a2f2f302f6261636b75703a107458b94bb4b64d37b9378b4264feeded'"],"ExecMode":"exec-internal","NumRows":1,"Age":18.742626}
I210409 04:39:41.156565 16906 jobs/registry.go:1175  [n1] 4543  BACKUP job 648389529063161857: stepping through state running with error: <nil>
I210409 04:39:41.159354 16907 9@util/log/event_log.go:32  [n1,job=648389529063161857,intExec=protectedts-GetRecord] 4544 ={"Timestamp":1617943181156937294,"EventType":"query_execute","Statement":"SELECT id, ts, meta_type, meta, spans, verified FROM \"\".system.protected_ts_records WHERE id = $1","User":"node","ApplicationName":"$ internal-protectedts-GetRecord","PlaceholderValues":["'\\x7458b94bb4b64d37b9378b4264feeded'"],"ExecMode":"exec-internal","NumRows":1,"Age":2.353807}
I210409 04:39:41.161755 16909 9@util/log/event_log.go:32  [intExec=protectedts-GetMetadata] 4545 ={"Timestamp":1617943181159913207,"EventType":"query_execute","Statement":"WITH current_meta AS (SELECT version, num_records, num_spans, total_bytes FROM \"\".system.protected_ts_meta UNION ALL SELECT 0 AS version, 0 AS num_records, 0 AS num_spans, 0 AS total_bytes ORDER BY version DESC LIMIT 1) SELECT version, num_records, num_spans, total_bytes FROM \"\".\"\".current_meta","User":"node","ApplicationName":"$ internal-protectedts-GetMetadata","ExecMode":"exec-internal","NumRows":1,"Age":1.781913,"FullTableScan":true}
I210409 04:39:41.164152 16910 9@util/log/event_log.go:32  [intExec=protectedts-GetMetadata] 4546 ={"Timestamp":1617943181159913207,"EventType":"query_execute","Statement":"WITH current_meta AS (SELECT version, num_records, num_spans, total_bytes FROM \"\".system.protected_ts_meta UNION ALL SELECT 0 AS version, 0 AS num_records, 0 AS num_spans, 0 AS total_bytes ORDER BY version DESC LIMIT 1) SELECT version, num_records, num_spans, total_bytes FROM \"\".\"\".current_meta","User":"node","ApplicationName":"$ internal-protectedts-GetMetadata","ExecMode":"exec-internal","NumRows":1,"Age":1.847843,"FullTableScan":true}
I210409 04:39:41.169959 16911 9@util/log/event_log.go:32  [intExec=protectedts-GetRecords] 4547 ={"Timestamp":1617943181159913207,"EventType":"query_execute","Statement":"SELECT id, ts, meta_type, meta, spans, verified FROM \"\".system.protected_ts_records","User":"node","ApplicationName":"$ internal-protectedts-GetRecords","ExecMode":"exec-internal","NumRows":1,"Age":4.580238,"FullTableScan":true}
I210409 04:39:41.181915 16912 9@util/log/event_log.go:32  [n1,job=648389529063161857,intExec=protectedts-MarkVerified] 4548 ={"Timestamp":1617943181171580551,"EventType":"query_execute","Statement":"UPDATE \"\".system.protected_ts_records SET verified = true WHERE id = $1 RETURNING true","User":"node","ApplicationName":"$ internal-protectedts-MarkVerified","PlaceholderValues":["'\\x7458b94bb4b64d37b9378b4264feeded'"],"ExecMode":"exec-internal","NumRows":1,"Age":10.263515}
I210409 04:39:41.201971 16964 ccl/backupccl/backup_processor.go:309  [n1,job=648389529063161857] 4549  sending ExportRequest for span /Table/75/{1-4} (attempt 1, priority normal)
I210409 04:39:41.202401 16964 ccl/storageccl/export.go:98  [n1,s1,r42/1:/{Table/75-Max}] 4550  export [/Table/75/1,/Table/75/4)
I210409 04:39:41.207515 16874 9@util/log/event_log.go:32  [n1,job=648389529063161857,intExec=log-job] 4551 ={"Timestamp":1617943181204367561,"EventType":"query_execute","Statement":"SELECT status, payload, progress, claim_session_id FROM \"\".system.jobs WHERE id = $1 FOR UPDATE","User":"root","ApplicationName":"$ internal-log-job","PlaceholderValues":["648389529063161857"],"ExecMode":"exec-internal","NumRows":1,"Age":3.048528}
I210409 04:39:41.212330 16876 9@util/log/event_log.go:32  [n1,job=648389529063161857,intExec=job-update] 4552 ={"Timestamp":1617943181204367561,"EventType":"query_execute","Statement":"UPDATE \"\".system.jobs SET progress = $2 WHERE id = $1","User":"root","ApplicationName":"$ internal-job-update","PlaceholderValues":["648389529063161857","'\\x0d0000803f1090b7d484acf0ef025200'"],"ExecMode":"exec-internal","NumRows":1,"Age":4.269901}
I210409 04:39:41.278532 16932 9@util/log/event_log.go:32  [n1,job=648389529063161857,intExec=protectedts-Release] 4553 ={"Timestamp":1617943181252468729,"EventType":"query_execute","Statement":"WITH current_meta AS (SELECT version, num_records, num_spans, total_bytes FROM \"\".system.protected_ts_meta UNION ALL SELECT 0 AS version, 0 AS num_records, 0 AS num_spans, 0 AS total_bytes ORDER BY version DESC LIMIT 1), record AS (SELECT id, num_spans AS record_spans, (length(spans) + length(meta_type)) + COALESCE(length(meta), 0) AS record_bytes FROM \"\".system.protected_ts_records WHERE id = $1), updated_meta AS (UPSERT INTO \"\".system.protected_ts_meta(version, num_records, num_spans, total_bytes) (SELECT version, num_records, num_spans, total_bytes FROM (SELECT version + 1 AS version, num_records - 1 AS num_records, num_spans - record_spans AS num_spans, total_bytes - record_bytes AS total_bytes FROM \"\".\"\".current_meta RIGHT JOIN \"\".\"\".record ON true)) RETURNING 1) DELETE FROM \"\".system.protected_ts_records AS r WHERE EXISTS (SELECT NULL FROM \"\".\"\".record WHERE r.id = record.id) RETURNING NULL","User":"node","Appli
I210409 04:39:41.278532 16932 9@util/log/event_log.go:32  [n1,job=648389529063161857,intExec=protectedts-Release] 4553 |cationName":"$ internal-protectedts-Release","PlaceholderValues":["'\\x7458b94bb4b64d37b9378b4264feeded'"],"ExecMode":"exec-internal","NumRows":1,"Age":25.942358,"FullTableScan":true}
I210409 04:39:41.321590 16479 9@util/log/event_log.go:32  [n1,job=648389529063161857,intExec=lookup-schedule-info] 4554 ={"Timestamp":1617943181313698139,"EventType":"query_execute","Statement":"SELECT created_by_id FROM \"\".system.jobs WHERE (id = $1) AND (created_by_type = $2)","User":"node","ApplicationName":"$ internal-lookup-schedule-info","PlaceholderValues":["648389529063161857","'crdb_schedule'"],"ExecMode":"exec-internal","Age":7.815166}
I210409 04:39:41.322443 16906 jobs/registry.go:1175  [n1] 4555  BACKUP job 648389529063161857: stepping through state succeeded with error: <nil>
I210409 04:39:41.333239 16701 9@util/log/event_log.go:32  [n1,intExec=log-job] 4556 ={"Timestamp":1617943181322540954,"EventType":"query_execute","Statement":"SELECT status, payload, progress, claim_session_id FROM \"\".system.jobs WHERE id = $1 FOR UPDATE","User":"root","ApplicationName":"$ internal-log-job","PlaceholderValues":["648389529063161857"],"ExecMode":"exec-internal","NumRows":1,"Age":10.612184}
I210409 04:39:41.346353 16950 9@util/log/event_log.go:32  [n1,intExec=job-update] 4557 ={"Timestamp":1617943181322540954,"EventType":"query_execute","Statement":"UPDATE \"\".system.jobs SET status = $2, payload = $3, progress = $4 WHERE id = $1","User":"root","ApplicationName":"$ internal-job-update","PlaceholderValues":["648389529063161857","'succeeded'","'\\x0a364241434b555020444154414241534520736368656d616368616e676520544f20276e6f64656c6f63616c3a2f2f302f6261636b7570271204726f6f7418bb80d084acf0ef0220bea7dc84acf0ef023206343f4048494b52360a00120a08e5bf90f783d885ba161a146e6f64656c6f63616c3a2f2f302f6261636b75703a107458b94bb4b64d37b9378b4264feeded'","'\\x0d0000803f10add2db84acf0ef025200'"],"ExecMode":"exec-internal","NumRows":1,"Age":12.839104}
I210409 04:39:41.347906 5136 9@util/log/event_log.go:32  [n1,client=127.0.0.1:38150,hostssl,user=root] 4558 ={"Timestamp":1617943181075825091,"EventType":"query_execute","Statement":"BACKUP DATABASE schemachange TO 'nodelocal://0/backup'","User":"root","ExecMode":"exec","NumRows":1,"Age":273.82922,"TxnCounter":7}
    schema_change_external_test.go:81: backup in /go/src/github.com/cockroachdb/cockroach/artifacts/TestWorkload246720409
I210409 04:39:41.359311 48 testutils/testcluster/testcluster.go:101  [-] 4559  TestCluster quiescing nodes
W210409 04:39:41.360077 1723 2@gossip/gossip.go:1491  [n3] 4561  no incoming or outgoing connections
W210409 04:39:41.360247 474 sql/sqlliveness/slinstance/slinstance.go:183  [n1] 4562  exiting heartbeat loop
W210409 04:39:41.360524 2074 jobs/registry.go:729  [-] 4563  canceling all adopted jobs due to stopper quiescing
W210409 04:39:41.360691 1614 jobs/registry.go:729  [-] 4564  canceling all adopted jobs due to stopper quiescing
E210409 04:39:41.360923 16952 kv/kvclient/kvcoord/txn_interceptor_committer.go:452  [-] 4565  making txn commit explicit failed for "unnamed" meta={id=f6c873fa pri=0.00800909 epo=0 ts=1617943181.322540954,0 min=1617943181.322540954,0 seq=6} lock=true stat=STAGING rts=1617943181.322540954,0 wto=false gul=1617943181.822540954,0 int=1 ifw=5: result is ambiguous (server shutdown)
W210409 04:39:41.361856 475 jobs/registry.go:729  [-] 4566  canceling all adopted jobs due to stopper quiescing
W210409 04:39:41.361955 14453 kv/kvserver/closedts/sidetransport/receiver.go:125  [n1] 4567  closed timestamps side-transport connection dropped from node: 2
W210409 04:39:41.362138 1613 sql/sqlliveness/slinstance/slinstance.go:183  [n2] 4568  exiting heartbeat loop
W210409 04:39:41.362392 14455 kv/kvserver/closedts/sidetransport/receiver.go:125  [n3] 4569  closed timestamps side-transport connection dropped from node: 2
W210409 04:39:41.362590 2239 kv/kvserver/closedts/sidetransport/receiver.go:125  [n3] 4570  closed timestamps side-transport connection dropped from node: 1
W210409 04:39:41.362653 2357 kv/kvserver/closedts/sidetransport/receiver.go:125  [n2] 4571  closed timestamps side-transport connection dropped from node: 1
W210409 04:39:41.359462 2073 sql/sqlliveness/slinstance/slinstance.go:183  [n3] 4560  exiting heartbeat loop
--- FAIL: TestWorkload (14.93s)
```
<details><summary>Reproduce</summary>
<p>
<p>To reproduce, try:

```bash
make stressrace TESTS=TestWorkload PKG=./pkg/ccl/testccl/workload/schemachange TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
</p>
<p>Parameters in this failure:

- GOFLAGS=-json
</p>
</p>
</details>

[See this test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestWorkload.*&sort=title+created&display=lastcommented+project)

<sub>[Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub>

[ccl/testccl/workload/schemachange.TestWorkload failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2864660&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=2864660&tab=artifacts#/) on [master @ c79c529648e968885105b7d516c9798aaca7d665](https://github.com/cockroachdb/cockroach/commits/c79c529648e968885105b7d516c9798aaca7d665):


```
1, 29, ui, 14
1, 29, users, 4
1, 29, web_sessions, 19
1, 29, zones, 5
50, 0, public, 29
51, 0, public, 29
52, 0, public, 29
52, 29, table20, 59
52, 29, table4, 56
52, 29, view28, 63
I210409 05:21:23.346218 7772 9@util/log/event_log.go:32  [n1,client=127.0.0.1:39906,hostssl,user=root,intExec=backup-lookup-tenant] 3286 ={"Timestamp":1617945683342812271,"EventType":"query_execute","Statement":"SELECT id, active, info FROM \"\".system.tenants","User":"root","ApplicationName":"$ internal-backup-lookup-tenant","ExecMode":"exec-internal","Age":0.912214,"FullTableScan":true}
I210409 05:21:23.364588 7773 9@util/log/event_log.go:32  [n1,client=127.0.0.1:39906,hostssl,user=root,intExec=job-row-insert] 3287 ={"Timestamp":1617945683342812271,"EventType":"query_execute","Statement":"INSERT INTO \"\".system.jobs(id, status, payload, progress, claim_session_id, claim_instance_id) VALUES ($1, $2, $3, $4, $5, $6)","User":"root","ApplicationName":"$ internal-job-row-insert","PlaceholderValues":["648397728469745665","'running'","'\\x0a364241434b555020444154414241534520736368656d616368616e676520544f20276e6f64656c6f63616c3a2f2f302f6261636b7570271204726f6f74320434383b3f52360a00120a08f490a6cceda086ba161a146e6f64656c6f63616c3a2f2f302f6261636b75703a102fe5526adecc48809703ac29c4ce02b0'","'\\x10dcebe2adb5f0ef025200'","'\\xde4d56586198451a87f394235c94fc96'","1"],"ExecMode":"exec-internal","NumRows":1,"Age":18.056648}
I210409 05:21:23.377968 7775 9@util/log/event_log.go:32  [n1,client=127.0.0.1:39906,hostssl,user=root,intExec=protectedts-protect] 3288 ={"Timestamp":1617945683342812271,"EventType":"query_execute","Statement":"WITH current_meta AS (SELECT version, num_records, num_spans, total_bytes FROM \"\".system.protected_ts_meta UNION ALL SELECT 0 AS version, 0 AS num_records, 0 AS num_spans, 0 AS total_bytes ORDER BY version DESC LIMIT 1), checks AS (SELECT new_version, new_num_records, new_num_spans, new_total_bytes, (((($1 > 0) AND (new_num_spans > $1)) OR (($2 > 0) AND (new_total_bytes > $2))) OR EXISTS (SELECT * FROM \"\".system.protected_ts_records WHERE id = $4)) AS failed FROM (SELECT version + 1 AS new_version, num_records + 1 AS new_num_records, num_spans + $3 AS new_num_spans, ((total_bytes + length($9)) + length($6)) + COALESCE(length($7), 0) AS new_total_bytes FROM \"\".\"\".current_meta)), updated_meta AS (UPSERT INTO \"\".system.protected_ts_meta(version, num_records, num_spans, total_bytes) (SELECT new_version, new_num_records, new_num_spans, new_total_bytes FROM \"\".\"\".checks WHERE NOT failed) RETURNING version, num_records, num_spans,
I210409 05:21:23.377968 7775 9@util/log/event_log.go:32  [n1,client=127.0.0.1:39906,hostssl,user=root,intExec=protectedts-protect] 3288 | total_bytes), new_record AS (INSERT INTO \"\".system.protected_ts_records(id, ts, meta_type, meta, num_spans, spans) (SELECT $4, $5, $6, $7, $8, $9 WHERE NOT EXISTS (SELECT * FROM \"\".\"\".checks WHERE failed)) RETURNING id) SELECT failed, num_spans AS prev_spans, total_bytes AS prev_total_bytes, version AS prev_version FROM \"\".\"\".checks, \"\".\"\".current_meta","User":"node","ApplicationName":"$ internal-protectedts-protect","PlaceholderValues":["32768","1048576","2","'\\x2fe5526adecc48809703ac29c4ce02b0'","'1617945683343018100.0000000000'","'jobs'","'\\x363438333937373238343639373435363635'","2","'\\x3a081a02c0892202c08e3a081a02c3892202c38a'"],"ExecMode":"exec-internal","NumRows":1,"Age":5.693575,"FullTableScan":true}
I210409 05:21:23.381188 7826 9@util/log/event_log.go:32  [n1,client=127.0.0.1:39906,hostssl,user=root,intExec=log-job] 3289 ={"Timestamp":1617945683379230111,"EventType":"query_execute","Statement":"SELECT status, payload, progress, claim_session_id FROM \"\".system.jobs WHERE id = $1 FOR UPDATE","User":"root","ApplicationName":"$ internal-log-job","PlaceholderValues":["648397728469745665"],"ExecMode":"exec-internal","NumRows":1,"Age":1.923181}
I210409 05:21:23.382820 7843 9@util/log/event_log.go:32  [n1,client=127.0.0.1:39906,hostssl,user=root,intExec=job-update] 3290 ={"Timestamp":1617945683379230111,"EventType":"query_execute","Statement":"UPDATE \"\".system.jobs SET status = $2, payload = $3 WHERE id = $1","User":"root","ApplicationName":"$ internal-job-update","PlaceholderValues":["648397728469745665","'running'","'\\x0a364241434b555020444154414241534520736368656d616368616e676520544f20276e6f64656c6f63616c3a2f2f302f6261636b7570271204726f6f7418b598e5adb5f0ef02320434383b3f52360a00120a08f490a6cceda086ba161a146e6f64656c6f63616c3a2f2f302f6261636b75703a102fe5526adecc48809703ac29c4ce02b0'"],"ExecMode":"exec-internal","NumRows":1,"Age":1.48393}
I210409 05:21:23.383759 7846 jobs/registry.go:1175  [n1] 3291  BACKUP job 648397728469745665: stepping through state running with error: <nil>
I210409 05:21:23.389347 7847 9@util/log/event_log.go:32  [n1,job=648397728469745665,intExec=protectedts-GetRecord] 3292 ={"Timestamp":1617945683383856593,"EventType":"query_execute","Statement":"SELECT id, ts, meta_type, meta, spans, verified FROM \"\".system.protected_ts_records WHERE id = $1","User":"node","ApplicationName":"$ internal-protectedts-GetRecord","PlaceholderValues":["'\\x2fe5526adecc48809703ac29c4ce02b0'"],"ExecMode":"exec-internal","NumRows":1,"Age":5.460827}
I210409 05:21:23.390962 7831 9@util/log/event_log.go:32  [intExec=protectedts-GetMetadata] 3293 ={"Timestamp":1617945683389605048,"EventType":"query_execute","Statement":"WITH current_meta AS (SELECT version, num_records, num_spans, total_bytes FROM \"\".system.protected_ts_meta UNION ALL SELECT 0 AS version, 0 AS num_records, 0 AS num_spans, 0 AS total_bytes ORDER BY version DESC LIMIT 1) SELECT version, num_records, num_spans, total_bytes FROM \"\".\"\".current_meta","User":"node","ApplicationName":"$ internal-protectedts-GetMetadata","ExecMode":"exec-internal","NumRows":1,"Age":1.321308,"FullTableScan":true}
I210409 05:21:23.392039 7832 9@util/log/event_log.go:32  [intExec=protectedts-GetMetadata] 3294 ={"Timestamp":1617945683389605048,"EventType":"query_execute","Statement":"WITH current_meta AS (SELECT version, num_records, num_spans, total_bytes FROM \"\".system.protected_ts_meta UNION ALL SELECT 0 AS version, 0 AS num_records, 0 AS num_spans, 0 AS total_bytes ORDER BY version DESC LIMIT 1) SELECT version, num_records, num_spans, total_bytes FROM \"\".\"\".current_meta","User":"node","ApplicationName":"$ internal-protectedts-GetMetadata","ExecMode":"exec-internal","NumRows":1,"Age":0.916501,"FullTableScan":true}
I210409 05:21:23.393134 7833 9@util/log/event_log.go:32  [intExec=protectedts-GetRecords] 3295 ={"Timestamp":1617945683389605048,"EventType":"query_execute","Statement":"SELECT id, ts, meta_type, meta, spans, verified FROM \"\".system.protected_ts_records","User":"node","ApplicationName":"$ internal-protectedts-GetRecords","ExecMode":"exec-internal","NumRows":1,"Age":0.867541,"FullTableScan":true}
I210409 05:21:23.394698 7834 9@util/log/event_log.go:32  [n1,job=648397728469745665,intExec=protectedts-MarkVerified] 3296 ={"Timestamp":1617945683393286033,"EventType":"query_execute","Statement":"UPDATE \"\".system.protected_ts_records SET verified = true WHERE id = $1 RETURNING true","User":"node","ApplicationName":"$ internal-protectedts-MarkVerified","PlaceholderValues":["'\\x2fe5526adecc48809703ac29c4ce02b0'"],"ExecMode":"exec-internal","NumRows":1,"Age":1.388858}
I210409 05:21:23.403045 7825 ccl/backupccl/backup_processor.go:309  [n1,job=648397728469745665] 3297  sending ExportRequest for span /Table/56/{1-6} (attempt 1, priority normal)
I210409 05:21:23.403114 7822 ccl/backupccl/backup_processor.go:309  [n1,job=648397728469745665] 3298  sending ExportRequest for span /Table/59/{1-2} (attempt 1, priority normal)
I210409 05:21:23.403295 7825 ccl/storageccl/export.go:98  [n1,s1,r38/1:/Table/5{6-9}] 3299  export [/Table/56/1,/Table/56/6)
I210409 05:21:23.403349 7822 ccl/storageccl/export.go:98  [n1,s1,r39/1:/{Table/59-Max}] 3300  export [/Table/59/1,/Table/59/2)
I210409 05:21:23.405875 7345 9@util/log/event_log.go:32  [n1,job=648397728469745665,intExec=log-job] 3301 ={"Timestamp":1617945683403876287,"EventType":"query_execute","Statement":"SELECT status, payload, progress, claim_session_id FROM \"\".system.jobs WHERE id = $1 FOR UPDATE","User":"root","ApplicationName":"$ internal-log-job","PlaceholderValues":["648397728469745665"],"ExecMode":"exec-internal","NumRows":1,"Age":1.934905}
I210409 05:21:23.407962 7795 9@util/log/event_log.go:32  [n1,job=648397728469745665,intExec=job-update] 3302 ={"Timestamp":1617945683403876287,"EventType":"query_execute","Statement":"UPDATE \"\".system.jobs SET progress = $2 WHERE id = $1","User":"root","ApplicationName":"$ internal-job-update","PlaceholderValues":["648397728469745665","'\\x0d0000003f10e4c8e6adb5f0ef025200'"],"ExecMode":"exec-internal","NumRows":1,"Age":1.917634}
I210409 05:21:23.411195 7890 9@util/log/event_log.go:32  [n1,job=648397728469745665,intExec=log-job] 3303 ={"Timestamp":1617945683409656908,"EventType":"query_execute","Statement":"SELECT status, payload, progress, claim_session_id FROM \"\".system.jobs WHERE id = $1 FOR UPDATE","User":"root","ApplicationName":"$ internal-log-job","PlaceholderValues":["648397728469745665"],"ExecMode":"exec-internal","NumRows":1,"Age":1.498385}
I210409 05:21:23.413004 7892 9@util/log/event_log.go:32  [n1,job=648397728469745665,intExec=job-update] 3304 ={"Timestamp":1617945683409656908,"EventType":"query_execute","Statement":"UPDATE \"\".system.jobs SET progress = $2 WHERE id = $1","User":"root","ApplicationName":"$ internal-job-update","PlaceholderValues":["648397728469745665","'\\x0d0000803f10f9f5e6adb5f0ef025200'"],"ExecMode":"exec-internal","NumRows":1,"Age":1.647928}
I210409 05:21:23.422390 7838 9@util/log/event_log.go:32  [n1,job=648397728469745665,intExec=get-table-statistics] 3305 ={"Timestamp":1617945683421876967,"EventType":"query_execute","Statement":"SELECT \"tableID\", \"statisticID\", name, \"columnIDs\", \"createdAt\", \"rowCount\", \"distinctCount\", \"nullCount\", histogram FROM \"\".system.table_statistics WHERE \"tableID\" = $1 ORDER BY \"createdAt\" DESC","User":"root","ApplicationName":"$ internal-get-table-statistics","PlaceholderValues":["63"],"ExecMode":"exec-internal","Age":1.39661,"TxnCounter":1}
I210409 05:21:23.429020 7839 9@util/log/event_log.go:32  [n1,job=648397728469745665,intExec=protectedts-Release] 3306 ={"Timestamp":1617945683424564106,"EventType":"query_execute","Statement":"WITH current_meta AS (SELECT version, num_records, num_spans, total_bytes FROM \"\".system.protected_ts_meta UNION ALL SELECT 0 AS version, 0 AS num_records, 0 AS num_spans, 0 AS total_bytes ORDER BY version DESC LIMIT 1), record AS (SELECT id, num_spans AS record_spans, (length(spans) + length(meta_type)) + COALESCE(length(meta), 0) AS record_bytes FROM \"\".system.protected_ts_records WHERE id = $1), updated_meta AS (UPSERT INTO \"\".system.protected_ts_meta(version, num_records, num_spans, total_bytes) (SELECT version, num_records, num_spans, total_bytes FROM (SELECT version + 1 AS version, num_records - 1 AS num_records, num_spans - record_spans AS num_spans, total_bytes - record_bytes AS total_bytes FROM \"\".\"\".current_meta RIGHT JOIN \"\".\"\".record ON true)) RETURNING 1) DELETE FROM \"\".system.protected_ts_records AS r WHERE EXISTS (SELECT NULL FROM \"\".\"\".record WHERE r.id = record.id) RETURNING NULL","User":"node","Appli
I210409 05:21:23.429020 7839 9@util/log/event_log.go:32  [n1,job=648397728469745665,intExec=protectedts-Release] 3306 |cationName":"$ internal-protectedts-Release","PlaceholderValues":["'\\x2fe5526adecc48809703ac29c4ce02b0'"],"ExecMode":"exec-internal","NumRows":1,"Age":4.39871,"FullTableScan":true}
I210409 05:21:23.431621 7907 9@util/log/event_log.go:32  [n1,job=648397728469745665,intExec=lookup-schedule-info] 3307 ={"Timestamp":1617945683429980107,"EventType":"query_execute","Statement":"SELECT created_by_id FROM \"\".system.jobs WHERE (id = $1) AND (created_by_type = $2)","User":"node","ApplicationName":"$ internal-lookup-schedule-info","PlaceholderValues":["648397728469745665","'crdb_schedule'"],"ExecMode":"exec-internal","Age":1.609749}
I210409 05:21:23.431784 7846 jobs/registry.go:1175  [n1] 3308  BACKUP job 648397728469745665: stepping through state succeeded with error: <nil>
I210409 05:21:23.433238 7908 9@util/log/event_log.go:32  [n1,intExec=log-job] 3309 ={"Timestamp":1617945683431836812,"EventType":"query_execute","Statement":"SELECT status, payload, progress, claim_session_id FROM \"\".system.jobs WHERE id = $1 FOR UPDATE","User":"root","ApplicationName":"$ internal-log-job","PlaceholderValues":["648397728469745665"],"ExecMode":"exec-internal","NumRows":1,"Age":1.372565}
I210409 05:21:23.435143 7910 9@util/log/event_log.go:32  [n1,intExec=job-update] 3310 ={"Timestamp":1617945683431836812,"EventType":"query_execute","Statement":"UPDATE \"\".system.jobs SET status = $2, payload = $3, progress = $4 WHERE id = $1","User":"root","ApplicationName":"$ internal-job-update","PlaceholderValues":["648397728469745665","'succeeded'","'\\x0a364241434b555020444154414241534520736368656d616368616e676520544f20276e6f64656c6f63616c3a2f2f302f6261636b7570271204726f6f7418b598e5adb5f0ef022092afe8adb5f0ef02320434383b3f52360a00120a08f490a6cceda086ba161a146e6f64656c6f63616c3a2f2f302f6261636b75703a102fe5526adecc48809703ac29c4ce02b0'","'\\x0d0000803f109da3e8adb5f0ef025200'"],"ExecMode":"exec-internal","NumRows":1,"Age":1.73899}
I210409 05:21:23.437177 4787 9@util/log/event_log.go:32  [n1,client=127.0.0.1:39906,hostssl,user=root] 3311 ={"Timestamp":1617945683342812271,"EventType":"query_execute","Statement":"BACKUP DATABASE schemachange TO 'nodelocal://0/backup'","User":"root","ExecMode":"exec","NumRows":1,"Age":94.41772,"TxnCounter":7}
    schema_change_external_test.go:81: backup in /go/src/github.com/cockroachdb/cockroach/artifacts/TestWorkload623514852
I210409 05:21:23.438080 37 testutils/testcluster/testcluster.go:101  [-] 3312  TestCluster quiescing nodes
W210409 05:21:23.438569 2495 kv/kvserver/closedts/sidetransport/receiver.go:125  [n3] 3313  closed timestamps side-transport connection dropped from node: 1
W210409 05:21:23.438678 392 jobs/registry.go:729  [-] 3314  canceling all adopted jobs due to stopper quiescing
W210409 05:21:23.438748 1845 jobs/registry.go:729  [-] 3315  canceling all adopted jobs due to stopper quiescing
W210409 05:21:23.439087 391 sql/sqlliveness/slinstance/slinstance.go:183  [n1] 3316  exiting heartbeat loop
W210409 05:21:23.439319 1502 jobs/registry.go:729  [-] 3317  canceling all adopted jobs due to stopper quiescing
W210409 05:21:23.439466 1844 sql/sqlliveness/slinstance/slinstance.go:183  [n3] 3318  exiting heartbeat loop
W210409 05:21:23.439585 1501 sql/sqlliveness/slinstance/slinstance.go:183  [n2] 3319  exiting heartbeat loop
E210409 05:21:23.439739 7912 kv/kvclient/kvcoord/txn_interceptor_committer.go:452  [-] 3320  making txn commit explicit failed for "unnamed" meta={id=6065969a pri=0.06193127 epo=0 ts=1617945683.431836812,0 min=1617945683.431836812,0 seq=6} lock=true stat=STAGING rts=1617945683.431836812,0 wto=false gul=1617945683.931836812,0 int=1 ifw=5: result is ambiguous (server shutdown)
W210409 05:21:23.440586 2021 kv/kvserver/closedts/sidetransport/receiver.go:125  [n2] 3321  closed timestamps side-transport connection dropped from node: 1
--- FAIL: TestWorkload (2.94s)
```
<details><summary>Reproduce</summary>
<p>
<p>To reproduce, try:

```bash
make stressrace TESTS=TestWorkload PKG=./pkg/ccl/testccl/workload/schemachange TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
</p>
<p>Parameters in this failure:

- GOFLAGS=-parallel=4
</p>
</p>
</details>

[See this test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestWorkload.*&sort=title+created&display=lastcommented+project)

<sub>[Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub>

[ccl/testccl/workload/schemachange.TestWorkload failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2866019&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=2866019&tab=artifacts#/) on [master @ 5be617bf6ee6a3b8bd9b6c60312f0bbf63d2c721](https://github.com/cockroachdb/cockroach/commits/5be617bf6ee6a3b8bd9b6c60312f0bbf63d2c721):


```
52, 29, enum81, 79
52, 29, table124, 86
52, 29, table178, 95
I210409 07:27:48.289073 23220 9@util/log/event_log.go:32  [n1,client=127.0.0.1:54978,hostssl,user=root,intExec=backup-lookup-tenant] 5711 ={"Timestamp":1617953268278111423,"EventType":"query_execute","Statement":"SELECT id, active, info FROM \"\".system.tenants","User":"root","ApplicationName":"$ internal-backup-lookup-tenant","ExecMode":"exec-internal","Age":2.808118,"FullTableScan":true}
I210409 07:27:48.293770 23221 9@util/log/event_log.go:32  [n1,client=127.0.0.1:54978,hostssl,user=root,intExec=job-row-insert] 5712 ={"Timestamp":1617953268278111423,"EventType":"query_execute","Statement":"INSERT INTO \"\".system.jobs(id, status, payload, progress, claim_session_id, claim_instance_id) VALUES ($1, $2, $3, $4, $5, $6)","User":"root","ApplicationName":"$ internal-job-row-insert","PlaceholderValues":["648422582813884417","'running'","'\\x0a364241434b555020444154414241534520736368656d616368616e676520544f20276e6f64656c6f63616c3a2f2f302f6261636b7570271204726f6f74320b34424344454f50565b5c5f52360a00120a08baf3b2d7cdfd87ba161a146e6f64656c6f63616c3a2f2f302f6261636b75703a10572b27a1dd414ecfb79b68f7fdaaa084'","'\\x10dfc6c6ced1f0ef025200'","'\\xff6e167e92be43cc89df2297c92d2b94'","1"],"ExecMode":"exec-internal","NumRows":1,"Age":3.304821}
I210409 07:27:48.353014 23223 9@util/log/event_log.go:32  [n1,client=127.0.0.1:54978,hostssl,user=root,intExec=protectedts-protect] 5713 ={"Timestamp":1617953268278111423,"EventType":"query_execute","Statement":"WITH current_meta AS (SELECT version, num_records, num_spans, total_bytes FROM \"\".system.protected_ts_meta UNION ALL SELECT 0 AS version, 0 AS num_records, 0 AS num_spans, 0 AS total_bytes ORDER BY version DESC LIMIT 1), checks AS (SELECT new_version, new_num_records, new_num_spans, new_total_bytes, (((($1 > 0) AND (new_num_spans > $1)) OR (($2 > 0) AND (new_total_bytes > $2))) OR EXISTS (SELECT * FROM \"\".system.protected_ts_records WHERE id = $4)) AS failed FROM (SELECT version + 1 AS new_version, num_records + 1 AS new_num_records, num_spans + $3 AS new_num_spans, ((total_bytes + length($9)) + length($6)) + COALESCE(length($7), 0) AS new_total_bytes FROM \"\".\"\".current_meta)), updated_meta AS (UPSERT INTO \"\".system.protected_ts_meta(version, num_records, num_spans, total_bytes) (SELECT new_version, new_num_records, new_num_spans, new_total_bytes FROM \"\".\"\".checks WHERE NOT failed) RETURNING version, num_records, num_spans,
I210409 07:27:48.353014 23223 9@util/log/event_log.go:32  [n1,client=127.0.0.1:54978,hostssl,user=root,intExec=protectedts-protect] 5713 | total_bytes), new_record AS (INSERT INTO \"\".system.protected_ts_records(id, ts, meta_type, meta, num_spans, spans) (SELECT $4, $5, $6, $7, $8, $9 WHERE NOT EXISTS (SELECT * FROM \"\".\"\".checks WHERE failed)) RETURNING id) SELECT failed, num_spans AS prev_spans, total_bytes AS prev_total_bytes, version AS prev_version FROM \"\".\"\".checks, \"\".\"\".current_meta","User":"node","ApplicationName":"$ internal-protectedts-protect","PlaceholderValues":["32768","1048576","2","'\\x572b27a1dd414ecfb79b68f7fdaaa084'","'1617953268278540730.0000000000'","'jobs'","'\\x363438343232353832383133383834343137'","2","'\\x3a081a02de892202de8c3a081a02e7892202e78d'"],"ExecMode":"exec-internal","NumRows":1,"Age":27.652344,"FullTableScan":true}
I210409 07:27:48.376149 23226 9@util/log/event_log.go:32  [n1,client=127.0.0.1:54978,hostssl,user=root,intExec=log-job] 5714 ={"Timestamp":1617953268366530899,"EventType":"query_execute","Statement":"SELECT status, payload, progress, claim_session_id FROM \"\".system.jobs WHERE id = $1 FOR UPDATE","User":"root","ApplicationName":"$ internal-log-job","PlaceholderValues":["648422582813884417"],"ExecMode":"exec-internal","NumRows":1,"Age":2.366619}
I210409 07:27:48.378293 23228 9@util/log/event_log.go:32  [n1,client=127.0.0.1:54978,hostssl,user=root,intExec=job-update] 5715 ={"Timestamp":1617953268366530899,"EventType":"query_execute","Statement":"UPDATE \"\".system.jobs SET status = $2, payload = $3 WHERE id = $1","User":"root","ApplicationName":"$ internal-job-update","PlaceholderValues":["648422582813884417","'running'","'\\x0a364241434b555020444154414241534520736368656d616368616e676520544f20276e6f64656c6f63616c3a2f2f302f6261636b7570271204726f6f7418fac5ccced1f0ef02320b34424344454f50565b5c5f52360a00120a08baf3b2d7cdfd87ba161a146e6f64656c6f63616c3a2f2f302f6261636b75703a10572b27a1dd414ecfb79b68f7fdaaa084'"],"ExecMode":"exec-internal","NumRows":1,"Age":1.923227}
I210409 07:27:48.402615 23248 jobs/registry.go:1175  [n1] 5716  BACKUP job 648422582813884417: stepping through state running with error: <nil>
I210409 07:27:48.405069 23249 9@util/log/event_log.go:32  [n1,job=648422582813884417,intExec=protectedts-GetRecord] 5717 ={"Timestamp":1617953268402745149,"EventType":"query_execute","Statement":"SELECT id, ts, meta_type, meta, spans, verified FROM \"\".system.protected_ts_records WHERE id = $1","User":"node","ApplicationName":"$ internal-protectedts-GetRecord","PlaceholderValues":["'\\x572b27a1dd414ecfb79b68f7fdaaa084'"],"ExecMode":"exec-internal","NumRows":1,"Age":2.271669}
I210409 07:27:48.409383 23268 9@util/log/event_log.go:32  [intExec=protectedts-GetMetadata] 5718 ={"Timestamp":1617953268405955654,"EventType":"query_execute","Statement":"WITH current_meta AS (SELECT version, num_records, num_spans, total_bytes FROM \"\".system.protected_ts_meta UNION ALL SELECT 0 AS version, 0 AS num_records, 0 AS num_spans, 0 AS total_bytes ORDER BY version DESC LIMIT 1) SELECT version, num_records, num_spans, total_bytes FROM \"\".\"\".current_meta","User":"node","ApplicationName":"$ internal-protectedts-GetMetadata","ExecMode":"exec-internal","NumRows":1,"Age":3.360901,"FullTableScan":true}
I210409 07:27:48.414897 23040 9@util/log/event_log.go:32  [intExec=protectedts-GetMetadata] 5719 ={"Timestamp":1617953268405955654,"EventType":"query_execute","Statement":"WITH current_meta AS (SELECT version, num_records, num_spans, total_bytes FROM \"\".system.protected_ts_meta UNION ALL SELECT 0 AS version, 0 AS num_records, 0 AS num_spans, 0 AS total_bytes ORDER BY version DESC LIMIT 1) SELECT version, num_records, num_spans, total_bytes FROM \"\".\"\".current_meta","User":"node","ApplicationName":"$ internal-protectedts-GetMetadata","ExecMode":"exec-internal","NumRows":1,"Age":5.308881,"FullTableScan":true}
I210409 07:27:48.424822 23284 9@util/log/event_log.go:32  [intExec=protectedts-GetRecords] 5720 ={"Timestamp":1617953268405955654,"EventType":"query_execute","Statement":"SELECT id, ts, meta_type, meta, spans, verified FROM \"\".system.protected_ts_records","User":"node","ApplicationName":"$ internal-protectedts-GetRecords","ExecMode":"exec-internal","NumRows":1,"Age":9.744027,"FullTableScan":true}
I210409 07:27:48.427965 23289 9@util/log/event_log.go:32  [n1,job=648422582813884417,intExec=protectedts-MarkVerified] 5721 ={"Timestamp":1617953268425628630,"EventType":"query_execute","Statement":"UPDATE \"\".system.protected_ts_records SET verified = true WHERE id = $1 RETURNING true","User":"node","ApplicationName":"$ internal-protectedts-MarkVerified","PlaceholderValues":["'\\x572b27a1dd414ecfb79b68f7fdaaa084'"],"ExecMode":"exec-internal","NumRows":1,"Age":2.291774}
I210409 07:27:48.440833 23309 ccl/backupccl/backup_processor.go:309  [n2] 5722  sending ExportRequest for span /Table/86/{1-4} (attempt 1, priority normal)
I210409 07:27:48.441246 23309 ccl/storageccl/export.go:98  [n2,s2,r42/2:/Table/8{6-9}] 5723  export [/Table/86/1,/Table/86/4)
I210409 07:27:48.441938 23309 ccl/backupccl/backup_processor.go:309  [n2] 5724  sending ExportRequest for span /Table/95/{1-5} (attempt 1, priority normal)
I210409 07:27:48.442119 23309 ccl/storageccl/export.go:98  [n2,s2,r59/2:/Table/9{5-7}] 5725  export [/Table/95/1,/Table/95/5)
I210409 07:27:48.448894 23292 9@util/log/event_log.go:32  [n1,job=648422582813884417,intExec=log-job] 5726 ={"Timestamp":1617953268445673962,"EventType":"query_execute","Statement":"SELECT status, payload, progress, claim_session_id FROM \"\".system.jobs WHERE id = $1 FOR UPDATE","User":"root","ApplicationName":"$ internal-log-job","PlaceholderValues":["648422582813884417"],"ExecMode":"exec-internal","NumRows":1,"Age":3.129488}
I210409 07:27:48.453550 23294 9@util/log/event_log.go:32  [n1,job=648422582813884417,intExec=job-update] 5727 ={"Timestamp":1617953268445673962,"EventType":"query_execute","Statement":"UPDATE \"\".system.jobs SET progress = $2 WHERE id = $1","User":"root","ApplicationName":"$ internal-job-update","PlaceholderValues":["648422582813884417","'\\x0d0000003f10eae3d0ced1f0ef025200'"],"ExecMode":"exec-internal","NumRows":1,"Age":2.81797}
I210409 07:27:48.459445 23273 9@util/log/event_log.go:32  [n1,job=648422582813884417,intExec=log-job] 5728 ={"Timestamp":1617953268457220077,"EventType":"query_execute","Statement":"SELECT status, payload, progress, claim_session_id FROM \"\".system.jobs WHERE id = $1 FOR UPDATE","User":"root","ApplicationName":"$ internal-log-job","PlaceholderValues":["648422582813884417"],"ExecMode":"exec-internal","NumRows":1,"Age":2.180456}
I210409 07:27:48.462104 23275 9@util/log/event_log.go:32  [n1,job=648422582813884417,intExec=job-update] 5729 ={"Timestamp":1617953268457220077,"EventType":"query_execute","Statement":"UPDATE \"\".system.jobs SET progress = $2 WHERE id = $1","User":"root","ApplicationName":"$ internal-job-update","PlaceholderValues":["648422582813884417","'\\x0d0000803f1084bed1ced1f0ef025200'"],"ExecMode":"exec-internal","NumRows":1,"Age":2.449186}
I210409 07:27:48.524095 23056 9@util/log/event_log.go:32  [n1,job=648422582813884417,intExec=protectedts-Release] 5730 ={"Timestamp":1617953268482219260,"EventType":"query_execute","Statement":"WITH current_meta AS (SELECT version, num_records, num_spans, total_bytes FROM \"\".system.protected_ts_meta UNION ALL SELECT 0 AS version, 0 AS num_records, 0 AS num_spans, 0 AS total_bytes ORDER BY version DESC LIMIT 1), record AS (SELECT id, num_spans AS record_spans, (length(spans) + length(meta_type)) + COALESCE(length(meta), 0) AS record_bytes FROM \"\".system.protected_ts_records WHERE id = $1), updated_meta AS (UPSERT INTO \"\".system.protected_ts_meta(version, num_records, num_spans, total_bytes) (SELECT version, num_records, num_spans, total_bytes FROM (SELECT version + 1 AS version, num_records - 1 AS num_records, num_spans - record_spans AS num_spans, total_bytes - record_bytes AS total_bytes FROM \"\".\"\".current_meta RIGHT JOIN \"\".\"\".record ON true)) RETURNING 1) DELETE FROM \"\".system.protected_ts_records AS r WHERE EXISTS (SELECT NULL FROM \"\".\"\".record WHERE r.id = record.id) RETURNING NULL","User":"node","Appli
I210409 07:27:48.524095 23056 9@util/log/event_log.go:32  [n1,job=648422582813884417,intExec=protectedts-Release] 5730 |cationName":"$ internal-protectedts-Release","PlaceholderValues":["'\\x572b27a1dd414ecfb79b68f7fdaaa084'"],"ExecMode":"exec-internal","NumRows":1,"Age":41.795685,"FullTableScan":true}
I210409 07:27:48.529039 23362 9@util/log/event_log.go:32  [n1,job=648422582813884417,intExec=lookup-schedule-info] 5731 ={"Timestamp":1617953268526465268,"EventType":"query_execute","Statement":"SELECT created_by_id FROM \"\".system.jobs WHERE (id = $1) AND (created_by_type = $2)","User":"node","ApplicationName":"$ internal-lookup-schedule-info","PlaceholderValues":["648422582813884417","'crdb_schedule'"],"ExecMode":"exec-internal","Age":2.490765}
I210409 07:27:48.529809 23248 jobs/registry.go:1175  [n1] 5732  BACKUP job 648422582813884417: stepping through state succeeded with error: <nil>
I210409 07:27:48.532477 23363 9@util/log/event_log.go:32  [n1,intExec=log-job] 5733 ={"Timestamp":1617953268530329254,"EventType":"query_execute","Statement":"SELECT status, payload, progress, claim_session_id FROM \"\".system.jobs WHERE id = $1 FOR UPDATE","User":"root","ApplicationName":"$ internal-log-job","PlaceholderValues":["648422582813884417"],"ExecMode":"exec-internal","NumRows":1,"Age":2.09495}
I210409 07:27:48.537051 23365 9@util/log/event_log.go:32  [n1,intExec=job-update] 5734 ={"Timestamp":1617953268530329254,"EventType":"query_execute","Statement":"UPDATE \"\".system.jobs SET status = $2, payload = $3, progress = $4 WHERE id = $1","User":"root","ApplicationName":"$ internal-job-update","PlaceholderValues":["648422582813884417","'succeeded'","'\\x0a364241434b555020444154414241534520736368656d616368616e676520544f20276e6f64656c6f63616c3a2f2f302f6261636b7570271204726f6f7418fac5ccced1f0ef0220ac8fd6ced1f0ef02320b34424344454f50565b5c5f52360a00120a08baf3b2d7cdfd87ba161a146e6f64656c6f63616c3a2f2f302f6261636b75703a10572b27a1dd414ecfb79b68f7fdaaa084'","'\\x0d0000803f1099f9d5ced1f0ef025200'"],"ExecMode":"exec-internal","NumRows":1,"Age":3.812517}
I210409 07:27:48.539231 5369 9@util/log/event_log.go:32  [n1,client=127.0.0.1:54978,hostssl,user=root] 5735 ={"Timestamp":1617953268278111423,"EventType":"query_execute","Statement":"BACKUP DATABASE schemachange TO 'nodelocal://0/backup'","User":"root","ExecMode":"exec","NumRows":1,"Age":261.2102,"TxnCounter":7}
    schema_change_external_test.go:81: backup in /go/src/github.com/cockroachdb/cockroach/artifacts/TestWorkload349588044
I210409 07:27:48.540849 33 testutils/testcluster/testcluster.go:101  [-] 5736  TestCluster quiescing nodes
W210409 07:27:48.541452 2095 jobs/registry.go:729  [-] 5737  canceling all adopted jobs due to stopper quiescing
W210409 07:27:48.541883 23090 kv/kvserver/intentresolver/intent_resolver.go:758  [-] 5738  failed to gc transaction record: could not GC completed transaction anchored at /Local/Range/Table/95/RangeDescriptor: node unavailable; try another peer
W210409 07:27:48.542002 1642 jobs/registry.go:729  [-] 5739  canceling all adopted jobs due to stopper quiescing
W210409 07:27:48.542060 16488 kv/kvserver/closedts/sidetransport/receiver.go:125  [n3] 5740  closed timestamps side-transport connection dropped from node: 2
W210409 07:27:48.542150 2224 kv/kvserver/closedts/sidetransport/receiver.go:125  [n3] 5741  closed timestamps side-transport connection dropped from node: 1
W210409 07:27:48.542281 1641 sql/sqlliveness/slinstance/slinstance.go:183  [n2] 5743  exiting heartbeat loop
W210409 07:27:48.542291 2094 sql/sqlliveness/slinstance/slinstance.go:183  [n3] 5744  exiting heartbeat loop
W210409 07:27:48.542277 9586 kv/kvserver/closedts/sidetransport/receiver.go:125  [n2] 5742  closed timestamps side-transport connection dropped from node: 3
W210409 07:27:48.542376 2261 kv/kvserver/closedts/sidetransport/receiver.go:125  [n2] 5745  closed timestamps side-transport connection dropped from node: 1
W210409 07:27:48.542911 2479 kv/kvserver/raft_transport.go:633  [n1] 5746  while processing outgoing Raft queue to node 2: EOF:
W210409 07:27:48.543252 2061 kv/kvserver/raft_transport.go:633  [n1] 5747  while processing outgoing Raft queue to node 3: EOF:
W210409 07:27:48.543561 2480 kv/kvserver/raft_transport.go:633  [n1] 5748  while processing outgoing Raft queue to node 3: EOF:
W210409 07:27:48.544012 16503 kv/kvserver/closedts/sidetransport/receiver.go:125  [n1] 5749  closed timestamps side-transport connection dropped from node: 2
W210409 07:27:48.544209 9576 kv/kvserver/closedts/sidetransport/receiver.go:125  [n1] 5750  closed timestamps side-transport connection dropped from node: 3
W210409 07:27:48.545382 2049 kv/kvserver/raft_transport.go:633  [n1] 5751  while processing outgoing Raft queue to node 2: EOF:
W210409 07:27:48.546285 548 jobs/registry.go:729  [-] 5752  canceling all adopted jobs due to stopper quiescing
W210409 07:27:48.546520 547 sql/sqlliveness/slinstance/slinstance.go:183  [n1] 5753  exiting heartbeat loop
--- FAIL: TestWorkload (21.54s)
```
<details><summary>Reproduce</summary>
<p>
<p>To reproduce, try:

```bash
make stressrace TESTS=TestWorkload PKG=./pkg/ccl/testccl/workload/schemachange TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
</p>
<p>Parameters in this failure:

- GOFLAGS=-json
</p>
</p>
</details>

[See this test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestWorkload.*&sort=title+created&display=lastcommented+project)

<sub>[Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub>

--- a/pkg/ccl/testccl/workload/schemachange/BUILD.bazel
+++ b/pkg/ccl/testccl/workload/schemachange/BUILD.bazel
@@ -18,7 +18,6 @@ go_test(
         "//pkg/security/securitytest",
         "//pkg/server",
         "//pkg/testutils/serverutils",
-        "//pkg/testutils/skip",
         "//pkg/testutils/sqlutils",
         "//pkg/testutils/testcluster",
         "//pkg/util/leaktest",

--- a/pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go
+++ b/pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go
@@ -21,7 +21,6 @@ import (
 	_ "github.com/cockroachdb/cockroach/pkg/ccl"
 	"github.com/cockroachdb/cockroach/pkg/ccl/multiregionccl/multiregionccltestutils"
 	"github.com/cockroachdb/cockroach/pkg/ccl/utilccl"
-	"github.com/cockroachdb/cockroach/pkg/testutils/skip"
 	"github.com/cockroachdb/cockroach/pkg/testutils/sqlutils"
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
 	"github.com/cockroachdb/cockroach/pkg/workload"
@@ -33,7 +32,6 @@ import (
 
 func TestWorkload(t *testing.T) {
 	defer leaktest.AfterTest(t)()
-	skip.WithIssue(t, 63064, "flaky test")
 	defer utilccl.TestingEnableEnterprise()()
 
 	dir := t.TempDir()

--- a/pkg/cmd/roachtest/tests/schemachange_random_load.go
+++ b/pkg/cmd/roachtest/tests/schemachange_random_load.go
@@ -57,7 +57,6 @@ func registerSchemaChangeRandomLoad(r registry.Registry) {
 			}
 			runSchemaChangeRandomLoad(ctx, t, c, maxOps, concurrency)
 		},
-		Skip: "flaky: https://github.com/cockroachdb/cockroach/issues/82133",
 	})
 
 	// Run a few representative scbench specs in CI.

--- a/pkg/workload/schemachange/error_screening.go
+++ b/pkg/workload/schemachange/error_screening.go
@@ -348,6 +348,17 @@ GROUP BY name;
 		}
 		// Next validate the uniqueness of both constraints and index expressions.
 		for constraintIdx, constraint := range constraints {
+			nonTupleConstraint := constraint[0]
+			if len(nonTupleConstraint) > 2 &&
+				nonTupleConstraint[0] == '(' &&
+				nonTupleConstraint[len(nonTupleConstraint)-1] == ')' {
+				nonTupleConstraint = nonTupleConstraint[1 : len(nonTupleConstraint)-1]
+			}
+			hasNullsQuery := strings.Builder{}
+			hasNullsQuery.WriteString("SELECT num_nulls(")
+			hasNullsQuery.WriteString(nonTupleConstraint)
+			hasNullsQuery.WriteString(") > 0 FROM (VALUES(")
+
 			tupleSelectQuery := strings.Builder{}
 			tupleSelectQuery.WriteString("SELECT array[(")
 			tupleSelectQuery.WriteString(constraint[0])
@@ -361,7 +372,7 @@ GROUP BY name;
 			}
 			collector := newExprColumnCollector(colInfo)
 			t.Walk(collector)
-			query.WriteString("SELECT EXISTS ( SELECT * FROM ")
+			query.WriteString("SELECT COUNT (*) > 0 FROM (SELECT * FROM ")
 			query.WriteString(tableName.String())
 			query.WriteString(" WHERE ")
 			query.WriteString(constraint[0])
@@ -370,30 +381,23 @@ GROUP BY name;
 			query.WriteString(constraint[0])
 			query.WriteString(" FROM (VALUES( ")
 			colIdx := 0
-			nullValueEncountered := false
 			for col := range collector.columnsObserved {
 				value := columnsToValues[col]
 				if colIdx != 0 {
 					query.WriteString(",")
 					columns.WriteString(",")
 					tupleSelectQuery.WriteString(",")
-				}
-				if value == "NULL" {
-					nullValueEncountered = true
-					break
+					hasNullsQuery.WriteString(",")
 				}
 				query.WriteString(value)
 				columns.WriteString(col)
+				hasNullsQuery.WriteString(value)
 				tupleSelectQuery.WriteString(value)
 				colIdx++
 			}
-			// Row is not comparable to others for unique constraints, since it has a
-			// NULL value.
-			// TODO (fqazi): In the future for check constraints we should evaluate
-			// things for them.
-			if nullValueEncountered {
-				continue
-			}
+			hasNullsQuery.WriteString(") ) AS T(")
+			hasNullsQuery.WriteString(columns.String())
+			hasNullsQuery.WriteString(")")
 			tupleSelectQuery.WriteString(") ) AS T(")
 			tupleSelectQuery.WriteString(columns.String())
 			tupleSelectQuery.WriteString(")")
@@ -404,24 +408,53 @@ GROUP BY name;
 			if err != nil {
 				return false, nil, err
 			}
-			exists, err := og.scanBool(ctx, evalTxn, query.String())
-			if err != nil {
+			// Detect if any null values exist.
+			handleEvalTxnError := func(err error) (bool, error) {
+				// No choice but to rollback, expression is malformed.
+				rollbackErr := evalTxn.Rollback(ctx)
+				if rollbackErr != nil {
+					return false, err
+				}
 				var pgErr *pgconn.PgError
 				if !errors.As(err, &pgErr) {
-					return false, nil, err
+					return false, err
 				}
 				// Only accept known error types for generated expressions.
 				if !isValidGenerationError(pgErr.Code) {
-					return false, nil, err
+					return false, err
 				}
 				generatedCodes = append(generatedCodes,
 					codesWithConditions{
 						{code: pgcode.MakeCode(pgErr.Code), condition: true},
 					}...,
 				)
+				return true, nil
+			}
+			hasNullValues, err := og.scanBool(ctx, evalTxn, hasNullsQuery.String())
+			if err != nil {
+				skipConstraint, err := handleEvalTxnError(err)
+				if err != nil {
+					return false, generatedCodes, err
+				}
+				if skipConstraint {
+					continue
+				}
+			}
+			// Skip if any null values exist in the expression
+			if hasNullValues {
 				continue
 			}
-			err = evalTxn.Rollback(ctx)
+			exists, err := og.scanBool(ctx, evalTxn, query.String())
+			if err != nil {
+				skipConstraint, err := handleEvalTxnError(err)
+				if err != nil {
+					return false, generatedCodes, err
+				}
+				if skipConstraint {
+					continue
+				}
+			}
+			err = evalTxn.Commit(ctx)
 			if err != nil {
 				return false, nil, err
 			}
@@ -1118,6 +1151,7 @@ func (og *operationGenerator) violatesFkConstraintsHelper(
 	}
 	// If the parent and child are the same table, then any rows in an existing
 	// insert may satisfy the same constraint.
+	var parentAndChildSameQueryColumns []string
 	if parentAndChildAreSameTable {
 		colsInfo, err := og.getTableColumns(ctx, tx, childTableName, false)
 		if err != nil {
@@ -1145,20 +1179,20 @@ func (og *operationGenerator) violatesFkConstraintsHelper(
 					return false, err
 				}
 			}
-			matches, err := og.scanBool(ctx, tx,
-				fmt.Sprintf("SELECT %s = %s", parentValueInSameInsert, childValue))
-			if err != nil {
-				return false, err
-			}
-			if matches {
-				return false, err
-			}
+			parentAndChildSameQueryColumns = append(parentAndChildSameQueryColumns,
+				fmt.Sprintf("%s = %s", parentValueInSameInsert, childValue))
 		}
 	}
+	checkSharedParentChildRows := ""
+	if len(parentAndChildSameQueryColumns) > 0 {
+		checkSharedParentChildRows = fmt.Sprintf("true = ANY (ARRAY [%s]) OR",
+			strings.Join(parentAndChildSameQueryColumns, ","))
+	}
 	return og.scanBool(ctx, tx, fmt.Sprintf(`
-	    SELECT count(*) = 0 from %s.%s
+	    SELECT %s count(*) = 0 from %s.%s
 	    WHERE %s = (%s)
-	`, parentTableSchema, parentTableName, parentColumn, childValue))
+	`,
+		checkSharedParentChildRows, parentTableSchema, parentTableName, parentColumn, childValue))
 }
 
 func (og *operationGenerator) columnIsInDroppingIndex(

--- a/pkg/workload/schemachange/error_screening.go
+++ b/pkg/workload/schemachange/error_screening.go
@@ -1079,12 +1079,13 @@ func (og *operationGenerator) violatesFkConstraints(
 			childColumnName := constraint[3]
 
 			// If self referential, there cannot be a violation.
-			if parentTableSchema == tableName.Schema() && parentTableName == tableName.Object() && parentColumnName == childColumnName {
+			parentAndChildAreSame := parentTableSchema == tableName.Schema() && parentTableName == tableName.Object()
+			if parentAndChildAreSame && parentColumnName == childColumnName {
 				continue
 			}
 
 			violation, err := og.violatesFkConstraintsHelper(
-				ctx, tx, columnNameToIndexMap, parentTableSchema, parentTableName, parentColumnName, childColumnName, row,
+				ctx, tx, columnNameToIndexMap, parentTableSchema, parentTableName, parentColumnName, tableName.String(), childColumnName, parentAndChildAreSame, row, rows,
 			)
 			if err != nil {
 				return false, err
@@ -1104,8 +1105,10 @@ func (og *operationGenerator) violatesFkConstraintsHelper(
 	ctx context.Context,
 	tx pgx.Tx,
 	columnNameToIndexMap map[string]int,
-	parentTableSchema, parentTableName, parentColumn, childColumn string,
+	parentTableSchema, parentTableName, parentColumn, childTableName, childColumn string,
+	parentAndChildAreSameTable bool,
 	row []string,
+	allRows [][]string,
 ) (bool, error) {
 
 	// If the value to insert in the child column is NULL and the column default is NULL, then it is not possible to have a fk violation.
@@ -1113,10 +1116,48 @@ func (og *operationGenerator) violatesFkConstraintsHelper(
 	if childValue == "NULL" {
 		return false, nil
 	}
-
+	// If the parent and child are the same table, then any rows in an existing
+	// insert may satisfy the same constraint.
+	if parentAndChildAreSameTable {
+		colsInfo, err := og.getTableColumns(ctx, tx, childTableName, false)
+		if err != nil {
+			return false, err
+		}
+		// Put values to be inserted into a column name to value map to simplify lookups.
+		columnsToValues := map[string]string{}
+		for name, idx := range columnNameToIndexMap {
+			columnsToValues[name] = row[idx]
+		}
+		colIdx := 0
+		for idx, colInfo := range colsInfo {
+			if colInfo.name == parentColumn {
+				colIdx = idx
+				break
+			}
+		}
+		for _, otherRow := range allRows {
+			parentValueInSameInsert := otherRow[columnNameToIndexMap[parentColumn]]
+			// If the parent column is generated, spend time to generate the value.
+			if colsInfo[colIdx].generated {
+				var err error
+				parentValueInSameInsert, err = og.generateColumn(ctx, tx, colsInfo[colIdx], columnsToValues)
+				if err != nil {
+					return false, err
+				}
+			}
+			matches, err := og.scanBool(ctx, tx,
+				fmt.Sprintf("SELECT %s = %s", parentValueInSameInsert, childValue))
+			if err != nil {
+				return false, err
+			}
+			if matches {
+				return false, err
+			}
+		}
+	}
 	return og.scanBool(ctx, tx, fmt.Sprintf(`
 	    SELECT count(*) = 0 from %s.%s
-	    WHERE %s = %s
+	    WHERE %s = (%s)
 	`, parentTableSchema, parentTableName, parentColumn, childValue))
 }
 

--- a/pkg/workload/schemachange/operation_generator.go
+++ b/pkg/workload/schemachange/operation_generator.go
@@ -253,8 +253,8 @@ var opWeights = []int{
 	setColumnNotNull:        1,
 	setColumnType:           0, // Disabled and tracked with #66662.
 	survive:                 1,
-	insertRow:               1, // Temporarily reduced because of #80820
-	validate:                2, // validate twice more often
+	insertRow:               10, // Temporarily reduced because of #80820
+	validate:                2,  // validate twice more often
 }
 
 // randOp attempts to produce a random schema change operation. It returns a

